---
title: "01_prepare_data"
author: "Garazi Martin Bideguren"
date: "2024-04-09"
output: html_document
---

# Prepare data

## Load data

Load the original data files outputted by the bioinformatic pipeline.

### Sample metadata

```{r load_sample_metadata, warning=FALSE, comments="", message=FALSE}
sample_metadata <- read.csv("data/Pyrenees_metadata_all_v2.csv",sep=",",header=T)%>%
  filter(EHI_number != "EHI00102") %>%
  filter(EHI_number != "EHI00182") %>%
  filter(EHI_number !="EHI00435")
```

### Read counts

```{r load_read_counts, warning=FALSE, comments="", message=FALSE}
read_counts <- read_tsv("data/DMB0113_counts.tsv") %>%
  rename(genome = 1) %>%
  select(-EHI00102, -EHI00182,-EHI00435) #remove samples
```

### Genome base hits

```{r load_genome_hits, warning=FALSE, comments="", message=FALSE}
genome_coverage <- read_tsv("data/DMB0113_coverage.tsv") %>%
  rename(genome = 1) %>%
  select(-EHI00102, -EHI00182,-EHI00435) #remove samples
```

### Genome taxonomy

```{r load_genome_metadata, warning=FALSE, comments="", message=FALSE}
genome_metadata <- read_tsv("data/DMB0113_mag_info.tsv") %>%
    rename(length=mag_size)
```

### Genome tree

```{r load_genome_tree, warning=FALSE, comments="", message=FALSE}
genome_tree <- read_tree("data/DMB0113.tree")
genome_tree$tip.label <- str_replace_all(genome_tree$tip.label,"'", "") #remove single quotes in MAG names
genome_tree <- keep.tip(genome_tree, tip=genome_metadata$genome) # keep only MAG tips
```

### Genome annotations

Distill annotations already into GIFTs

```{r genome_annotations, message=FALSE}
genome_gifts_raw="data/GIFTs.tsv"
genome_gifts <- read.table(genome_gifts_raw,header=T, sep="\t", row.names=1)
```

## Create working objects

Transform the original data files into working objects for downstream analyses.

### Filter reads by coverage

```{r filter_coverage, warning=FALSE, comments="", message=FALSE}
min_coverage=0.3
read_counts_filt <- genome_coverage %>%
  mutate(across(where(is.numeric), ~ ifelse(. > min_coverage, 1, 0))) %>%
  mutate(across(-1, ~ . * read_counts[[cur_column()]])) 
```

### Transform reads into genome counts

```{r calculate_genome_counts_unfiltered, warning=FALSE, comments="", message=FALSE}
readlength=150
genome_counts <- read_counts %>%
  mutate(across(where(is.numeric), ~ . / (genome_metadata$length / readlength) ))
```

```{r calculate_genome_counts_filtered, warning=FALSE, comments="", message=FALSE}
readlength=150
genome_counts_filt <- read_counts_filt %>%
  mutate(across(where(is.numeric), ~ . / (genome_metadata$length / readlength) ))
```


## Prepare color scheme

[AlberdiLab](www.alberdilab.dk) projects use unified color schemes developed for the [Earth Hologenome Initiative](www.earthhologenome.org), to facilitate figure interpretation.

```{r get_ehi_colors, warning=FALSE, comments="", message=FALSE}
phylum_colors <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
    right_join(genome_metadata, by=join_by(phylum == phylum)) %>%
    arrange(match(genome, genome_tree$tip.label)) %>%
    select(phylum, colors) %>% 
    unique() %>%
    arrange(phylum) %>%
    pull(colors, name=phylum)
```

## Wrap working objects

All working objects are wrapped into a single Rdata object to facilitate downstream usage.

```{r wrap_working_objects, warning=FALSE, comments="", message=FALSE}
save(sample_metadata, 
     genome_metadata, 
     read_counts, 
     genome_counts, 
     genome_counts_filt, 
     genome_tree,
    genome_gifts, 
     phylum_colors,
     file = "data/data.Rdata")
```