# Diversity analyses

```{r load_data_community_1}
load("data/data.Rdata")
```

## Alpha diversity

```{r alpha_div, comment="", message=FALSE, warning=FALSE}
# Calculate Hill numbers
richness <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 0) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(richness = 1) %>%
  rownames_to_column(var = "sample")

neutral <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(neutral = 1) %>%
  rownames_to_column(var = "sample")

phylogenetic <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1, tree = genome_tree) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(phylogenetic = 1) %>%
  rownames_to_column(var = "sample")

# Aggregate basal GIFT into elements
#Get list of present MAGs
present_MAGs <- genome_counts_filt %>%
    column_to_rownames(var = "genome") %>%
		filter(rowSums(.[, -1]) != 0) %>%
		rownames()

#Align KEGG annotations with present MAGs and remove all-zero and all-one traits
present_MAGs <- present_MAGs[present_MAGs %in% rownames(genome_gifts)]
genome_gifts_filt <- genome_gifts[present_MAGs,] %>%
			select_if(~!all(. == 0)) %>%  #remove all-zero modules
			select_if(~!all(. == 1)) #remove all-one modules

#Filter count table to only contain present MAGs after KEGG filtering
genome_counts_filt_filt <- genome_counts_filt %>%
  column_to_rownames(var = "genome")

genome_counts_filt_filt <- genome_counts_filt_filt[present_MAGs,]


dist <- genome_gifts_filt %>%
  to.elements(., GIFT_db) %>%
  traits2dist(., method = "gower")

functional <- genome_counts_filt_filt %>%
  #column_to_rownames(var = "genome") %>%
  #dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1, dist = dist) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(functional = 1) %>%
  rownames_to_column(var = "sample") %>%
  mutate(functional = if_else(is.nan(functional), 1, functional))

# Merge all metrics
alpha_div <- richness %>%
  full_join(neutral, by = join_by(sample == sample)) %>%
  full_join(phylogenetic, by = join_by(sample == sample)) %>%
  full_join(functional, by = join_by(sample == sample))
```
```{r alpha_div_summary, comment="", message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
  pivot_longer(-sample, names_to = "alpha", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == EHI_number)) %>%
    group_by(alpha)%>%
    summarise(
              Aisa_mean=mean(value[Transect=="Aisa"], na.rm=T),
              Aisa_sd=sd(value[Transect=="Aisa"], na.rm=T),
              Aran_mean=mean(value[Transect=="Aran"], na.rm=T),
              Aran_sd=sd(value[Transect=="Aran"], na.rm=T),
              Sentein_mean=mean(value[Transect=="Sentein"], na.rm=T),
              Sentein_sd=sd(value[Transect=="Sentein"], na.rm=T),
              Tourmalet_mean=mean(value[Transect=="Tourmalet"], na.rm=T),
              Tourmalet_sd=sd(value[Transect=="Tourmalet"], na.rm=T))%>%
    mutate(
           Aisa=str_c(round(Aisa_mean,2),"±",round(Aisa_sd,2)),
           Aran=str_c(round(Aran_mean,2),"±",round(Aran_sd,2)),
           Sentein=str_c(round(Sentein_mean,2),"±",round(Sentein_sd,2)),
           Tourmalet=str_c(round(Tourmalet_mean,2),"±",round(Tourmalet_sd,2))) %>% 
    arrange(-Aisa_mean) %>% 
    dplyr::select(alpha,Aisa,Aran,Sentein,Tourmalet) %>% 
    tt()
```

```{r alpha_div_plot, comment="", message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == EHI_number)) %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic","functional"))) %>%
      ggplot(aes(y = value, x = Transect, group=Transect, color=Transect, fill=Transect)) +
      geom_boxplot(outlier.shape = NA) +
      geom_jitter(alpha=0.5) +
      scale_color_manual(name="Transect",
          breaks=c("Aisa","Aran","Sentein","Tourmalet"),
          labels=c("Aisa","Aran","Sentein","Tourmalet"),
          values=c("#e5bd5b", "#6b7398","#e2815a", "#876b96")) +
      scale_fill_manual(name="Transect",
          breaks=c("Aisa","Aran","Sentein","Tourmalet"),
          labels=c("Aisa","Aran","Sentein","Tourmalet"),
          values=c("#e5bd5b50", "#6b739850","#e2815a50", "#876b9650")) +
      facet_wrap(. ~ metric, scales = "free", ncol=4) +
      coord_cartesian(xlim = c(1, NA)) +
      stat_compare_means(size=2) +
      theme_classic() +
      theme(
        strip.background = element_blank(),
        panel.grid.minor.x = element_line(size = .1, color = "grey"),
        axis.text.x = element_text(angle = 45, hjust = 1)
      ) +
  ylab("Alpha diversity")      
      
```

### Regression plots

#### Richness diversity

```{r alpha_div_richness_plot, message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
columns <- c("richness","neutral","phylo","func","mapped","total")
alpha_div %>%
			select(sample,richness) %>%
			pivot_longer(-sample, names_to = "data", values_to = "value") %>%
			mutate(data = factor(data, levels = columns))	%>%
			left_join(sample_metadata, by = join_by(sample == EHI_number)) %>%
      ggplot(aes(x = Elevation, y = value)) +
        geom_point() +
        geom_smooth(method = "loess") +
        stat_poly_eq()+
        facet_wrap(~ factor(Transect))+
        labs(x = "Elevation (m)")
```

#### Neutral diversity

```{r alpha_div_neutral_plot, message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
			select(sample,neutral) %>%
			pivot_longer(-sample, names_to = "data", values_to = "value") %>%
			mutate(data = factor(data, levels = columns))	%>%
			left_join(sample_metadata, by = join_by(sample == EHI_number))  %>%
      ggplot(aes(x = Elevation, y = value)) +
        geom_point() +
        geom_smooth(method = "loess") +
        stat_poly_eq()+
        facet_wrap(~ factor(Transect))+
        labs(x = "Elevation (m)")
```

#### Phylogenetic diversity

```{r alpha_div_phylo_plot, message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
			select(sample,phylogenetic) %>%
			pivot_longer(-sample, names_to = "data", values_to = "value") %>%
			mutate(data = factor(data, levels = columns))	%>%
			left_join(sample_metadata, by = join_by(sample == EHI_number)) %>%
      ggplot(aes(x = Elevation, y = value)) +
        geom_point() +
        geom_smooth(method = "loess") +
        stat_poly_eq()+
        facet_wrap(~ factor(Transect))+
        labs(x = "Elevation (m)")
```

#### Functional diversities

```{r alpha_div_func_plot, message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
			select(sample,functional) %>%
			pivot_longer(-sample, names_to = "data", values_to = "value") %>%
			mutate(data = factor(data, levels = columns))	%>%
			left_join(sample_metadata, by = join_by(sample == EHI_number))  %>%
      ggplot(aes(x = Elevation, y = value)) +
        geom_point() +
        geom_smooth(method = "loess") +
        stat_poly_eq()+
        facet_wrap(~ factor(Transect))+
        labs(x = "Elevation (m)")
```

### Mixed models

#### Richness diversity

```{r mixed_models_richness, echo=TRUE, message=FALSE, warning=FALSE, eval=FALSE}
richness_cor<-alpha_div %>%
  left_join(sample_metadata, by = join_by(sample == EHI_number)) %>%
  lmerTest::lmer(richness ~ log(sequencing_depth) + Elevation*Transect + (1 | Sampling_point), data = ., REML = FALSE) %>%
  broom.mixed::tidy() %>%
  tt()
```

```{r mixed_models_richness_df, echo=FALSE, message=FALSE, warning=FALSE,fig.height=4, fig.width=8, fig.fullwidth=TRUE}
richness_cor %>%
  paged_table()
```

#### Neutral diversity

```{r mixed_models_neutral, echo=TRUE, message=FALSE, warning=FALSE, eval=FALSE}
neutral_cor<-alpha_div %>%
  left_join(sample_metadata, by = join_by(sample == EHI_number)) %>%
  lmerTest::lmer(neutral ~ log(sequencing_depth) + Elevation*Transect + (1|Sampling_point), data = ., REML = FALSE) %>%
  broom.mixed::tidy() %>%
  tt()
```

```{r mixed_models_neutral_df, echo=FALSE, message=FALSE, warning=FALSE,fig.height=4, fig.width=8, fig.fullwidth=TRUE}
neutral_cor %>%
  paged_table()
```

#### Phylogenetic diversity

```{r mixed_models_phylo, echo=TRUE, message=FALSE, warning=FALSE, eval=FALSE}
phylo_cor<-alpha_div %>%
  left_join(sample_metadata, by = join_by(sample == EHI_number)) %>%
  lmerTest::lmer(phylogenetic ~ log(sequencing_depth) + Elevation*Transect + (1 | Sampling_point), data = ., REML = FALSE) %>%
  broom.mixed::tidy() %>%
  tt()
```

```{r mixed_models_phylo_df, echo=FALSE, message=FALSE, warning=FALSE,fig.height=4, fig.width=8, fig.fullwidth=TRUE}
phylo_cor %>%
  paged_table()
```

#### Functional diversities

```{r mixed_models_func, echo=TRUE, message=FALSE, warning=FALSE, eval=FALSE}
func_cor<-alpha_div %>%
  left_join(sample_metadata, by = join_by(sample == EHI_number)) %>%
  lmerTest::lmer(functional ~ log(sequencing_depth) + Elevation*Transect + (1 | Sampling_point), data = ., REML = FALSE) %>%
  broom.mixed::tidy() %>%
  tt()
```

```{r mixed_models_func_df, echo=FALSE, message=FALSE, warning=FALSE,fig.height=4, fig.width=8, fig.fullwidth=TRUE}
func_cor %>%
  paged_table()
```

## Beta diversity

```{r beta_div, comment="", message=FALSE, warning=FALSE, eval=FALSE}
beta_q0n <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  hillpair(., q = 0)

beta_q1n <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  hillpair(., q = 1)

beta_q1p <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  hillpair(., q = 1, tree = genome_tree)

beta_q1f <- genome_counts_filt_filt %>%
  #column_to_rownames(., "genome") %>%
  hillpair(., q = 1, dist = dist)
```

### Beta diversity per individual

#### Richness

```{r beta_div_richness_nmds, echo=TRUE, results = FALSE}
beta_q0n_nmds <- beta_q0n$S %>%
                metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
                vegan::scores() %>%
                as_tibble(., rownames = "sample") %>%
                left_join(sample_metadata, by = join_by(sample == EHI_number))
```

```{r beta_div_richness_centroids, echo=TRUE}
group_n <- length(unique(beta_q0n_nmds$Transect))
beta_q0n_nmds$Elevation<-as.character(beta_q0n_nmds$Elevation)

beta_q0n_nmds %>%
            group_by(Elevation) %>%
            mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
            mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
            ungroup() %>%
            ggplot(., aes(x=NMDS1,y=NMDS2, color=Elevation)) +
                #scale_color_manual(values=beta_colors[c(1:group_n)]) +
                geom_point(size=2) +
                geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
                theme_classic() +
                facet_wrap(.~Transect, scale="free") +
                theme(legend.position="right", legend.box="vertical") +
                guides(color=guide_legend(title="Elevation"))
```

```{r beta_div_richness_permutest, echo=TRUE, results = TRUE}
betadisper(beta_q0n$S, sample_metadata$Transect) %>% permutest(., pairwise=TRUE) 
```

```{r beta_div_richness_adonis, echo=TRUE, results = TRUE}
adonis2(beta_q0n$S ~ Elevation+Transect+Sampling_point, 
        data = sample_metadata %>% arrange(match(sample,labels(beta_q0n$S))), 
        permutations = 999,
        by="terms") %>%
        broom::tidy() %>%
        tt()
```

#### Neutral

```{r beta_div_neutral_nmds, echo=TRUE, results = FALSE}
beta_q1n_nmds <- beta_q1n$S %>%
                metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
                vegan::scores() %>%
                as_tibble(., rownames = "sample") %>%
                left_join(sample_metadata, by = join_by(sample == EHI_number))
```

```{r beta_div_neutral_centroids, echo=TRUE}
group_n <- length(unique(beta_q1n_nmds$Transect))
beta_q1n_nmds$Elevation<-as.character(beta_q1n_nmds$Elevation)

beta_q1n_nmds %>%
            group_by(Elevation) %>%
            mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
            mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
            ungroup() %>%
            ggplot(., aes(x=NMDS1,y=NMDS2, color=Elevation)) +
                #scale_color_manual(values=beta_colors[c(1:group_n)]) +
                geom_point(size=2) +
                geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
                theme_classic() +
                facet_wrap(.~Transect, scale="free") +
                theme(legend.position="right", legend.box="vertical") +
                guides(color=guide_legend(title="Elevation"))
```

```{r beta_div_neutral_permutest, echo=TRUE, results = TRUE}
betadisper(beta_q1n$S, sample_metadata$Transect) %>% permutest(., pairwise=TRUE) 
```

```{r beta_div_neutral_adonis, echo=TRUE, results = TRUE}
adonis2(beta_q1n$S ~ Elevation*Transect+Sampling_point, 
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1n$S))), 
        permutations = 999,
        by="terms") %>%
        broom::tidy() %>%
        tt()
```

#### Phylogenetic

```{r beta_div_phylo_nmds, echo=TRUE, results = FALSE}
beta_q1p_nmds <- beta_q1p$S %>%
                metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
                vegan::scores() %>%
                as_tibble(., rownames = "sample") %>%
                left_join(sample_metadata, by = join_by(sample == EHI_number))
```

```{r beta_div_phylo_centroids, echo=TRUE}
beta_q1p_nmds$Elevation<-as.character(beta_q1p_nmds$Elevation)

beta_q1p_nmds %>%
            group_by(Elevation) %>%
            mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
            mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
            ungroup() %>%
            ggplot(., aes(x=NMDS1,y=NMDS2, color=Elevation)) +
                #scale_color_manual(values=beta_colors[c(1:group_n)]) +
                geom_point(size=2) +
                geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
                theme_classic() +
                facet_wrap(.~Transect, scale="free") +
                theme(legend.position="right", legend.box="vertical") +
                guides(color=guide_legend(title="Elevation"))
```

```{r beta_div_phylo_permutest, echo=TRUE, results = TRUE}
betadisper(beta_q1p$S, sample_metadata$Transect) %>% permutest(., pairwise=TRUE) 
```

```{r beta_div_phylo_adonis, echo=TRUE, results = TRUE}
adonis2(beta_q1p$S ~ Elevation+Transect+Sampling_point, 
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1p$S))), 
        permutations = 999,
        by="terms") %>%
        broom::tidy() %>%
        tt()
```

#### Functional

```{r beta_div_func_nmds, echo=TRUE, results = FALSE}
beta_q1f_nmds <- beta_q1f$S %>%
                metaMDS(.,trymax = 500, k=2, verbosity=FALSE) %>%
                vegan::scores() %>%
                as_tibble(., rownames = "sample") %>%
                left_join(sample_metadata, by = join_by(sample == EHI_number))
```

```{r beta_div_func_centroids, echo=TRUE}
group_n <- length(unique(beta_q1f_nmds$Transect))
beta_q1f_nmds$Elevation<-as.character(beta_q1f_nmds$Elevation)


beta_q1f_nmds %>%
            group_by(Elevation) %>%
            mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
            mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
            ungroup() %>%
            ggplot(., aes(x=NMDS1,y=NMDS2, color=Elevation)) +
                #scale_color_manual(values=beta_colors[c(1:group_n)]) +
                geom_point(size=2) +
                geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
                theme_classic() +
                facet_wrap(.~Transect, scale="free") +
                theme(legend.position="right", legend.box="vertical") +
                guides(color=guide_legend(title="Elevation"))
```

```{r beta_div_func_permutest, echo=TRUE, results = TRUE}
betadisper(beta_q1f$S, sample_metadata$Transect) %>% permutest(., pairwise=TRUE) 
```

```{r beta_div_func_adonis, echo=TRUE, results = TRUE}
adonis2(beta_q1f$S ~ Elevation+Transect+Sampling_point, 
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1f$S))), 
        permutations = 999,
        by="terms") %>%
        broom::tidy() %>%
        tt()
```

### Beta diversity per population

```{r beta_div_transects, comment="", message=FALSE, warning=FALSE, eval=FALSE}
richness_beta <- genome_counts_filt %>%
  pivot_longer(!genome,names_to="sample",values_to = "counts") %>%
  left_join(sample_metadata, by = join_by(sample == EHI_number)) %>%
  mutate(Sampling_point=factor(Sampling_point)) %>% 
  group_by(Sampling_point) %>% 
  group_split() %>% 
  map_dbl(., ~ .x %>%
    select(genome, sample, counts) %>%
    pivot_wider(names_from = sample, values_from = counts) %>%
    column_to_rownames(var = "genome") %>%
    tss() %>%
    as.data.frame() %>%
    hilldiss(data=., metric="S", q = 0)
  )
names(richness_beta)  <- unique(sample_metadata$Sampling_point) %>% sort()


neutral_beta <- genome_counts_filt %>%
  pivot_longer(!genome,names_to="sample",values_to = "counts") %>%
  left_join(sample_metadata, by = join_by(sample == EHI_number)) %>%
  mutate(Sampling_point=factor(Sampling_point)) %>% 
  group_by(Sampling_point) %>% 
  group_split() %>% 
  map_dbl(., ~ .x %>%
    select(genome, sample, counts) %>%
    pivot_wider(names_from = sample, values_from = counts) %>%
    column_to_rownames(var = "genome") %>%
    tss() %>%
    as.data.frame() %>%
    hilldiss(data=., metric="S", q = 1)
  )
names(neutral_beta)  <- unique(sample_metadata$Sampling_point) %>% sort()

phylogenetic_beta <- genome_counts_filt %>%
  pivot_longer(!genome,names_to="sample",values_to = "counts") %>%
  left_join(sample_metadata, by = join_by(sample == EHI_number)) %>%
  mutate(Sampling_point=factor(Sampling_point)) %>% 
  group_by(Sampling_point) %>% 
  group_split() %>% 
  map_dbl(., ~ .x %>%
    select(genome, sample, counts) %>%
    pivot_wider(names_from = sample, values_from = counts) %>%
    column_to_rownames(var = "genome") %>%
    tss() %>%
    as.data.frame() %>%
    hilldiss(data=., metric="S", q = 1, tree = genome_tree)
  )
names(phylogenetic_beta)  <- unique(sample_metadata$Sampling_point) %>% sort()


genome_counts_filt_filt <- tibble::rownames_to_column(genome_counts_filt_filt, var = "genome")

functional_beta <- genome_counts_filt_filt %>%
  tibble::rownames_to_column(var = "genome") %>%
  pivot_longer(!genome,names_to="sample",values_to = "counts") %>%
  left_join(sample_metadata, by = join_by(sample == EHI_number)) %>%
  mutate(Sampling_point=factor(Sampling_point)) %>% 
  group_by(Sampling_point) %>% 
  group_split() %>% 
  map_dbl(., ~ .x %>%
    select(genome, sample, counts) %>%
    pivot_wider(names_from = sample, values_from = counts) %>%
    column_to_rownames(var = "genome") %>%
    tss() %>%
    as.data.frame() %>%
    hilldiss(data=., metric="S", q = 1, dist = dist)
  )
names(functional_beta)  <- unique(sample_metadata$Sampling_point) %>% sort()

# Merge all metrics
beta_div <- bind_rows(richness_beta,neutral_beta,phylogenetic_beta,functional_beta) %>%
  t() %>% 
  as.data.frame() %>% 
  rownames_to_column(var="Sampling_point") %>% 
  as_tibble() %>% 
  rename("richness"=V1,"neutral"=V2,"phylogenetic"=V3,"functional"=V4)
```

```{r beta_div_boxplot, comment="", message=FALSE, warning=FALSE, fig.height=5, fig.width=10, fig.fullwidth=TRUE}
beta_div %>%
  pivot_longer(-Sampling_point, names_to = "metric", values_to = "value") %>%
  left_join(sample_metadata %>% select(Transect,Sampling_point, Elevation) %>% unique, by = "Sampling_point") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic","functional"))) %>%
      ggplot(aes(y = value, x = Transect,group=Transect, fill=Transect)) +
      geom_boxplot(outlier.shape = NA) +
      geom_jitter(aes(color = Elevation), alpha = 0.5, size = 2) +
      scale_colour_gradient(low="#a7f0ba", high="#044319") +
      scale_fill_manual(name="Transect",
          breaks=c("Aisa","Aran","Sentein","Tourmalet"),
          labels=c("Aisa","Aran","Sentein","Tourmalet"),
          values=c("#e5bd5b50", "#6b739850","#e2815a50", "#876b9650")) +
      facet_wrap(. ~ metric, scales = "free", ncol=4) +
      coord_cartesian(xlim = c(1, NA)) +
      stat_compare_means(size=2) +
      theme_classic() +
      theme(
        strip.background = element_blank(),
        panel.grid.minor.x = element_line(size = .1, color = "grey"),
        axis.text.x = element_text(angle = 45, hjust = 1)
      ) + 
  ylab("Beta diversity")

```

```{r beta_div_boxplot_1, comment="", results=FALSE, message=FALSE, warning=FALSE, fig.height=5, fig.width=10, fig.fullwidth=TRUE}
bxp1<-beta_div %>%
  pivot_longer(-Sampling_point, names_to = "metric", values_to = "value") %>%
  left_join(sample_metadata %>% select(Transect,Sampling_point, Elevation) %>% unique, by = "Sampling_point") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic","functional"))) %>%
      ggplot(aes(y = value, x = Elevation ,group=Elevation, colour=Transect)) +
      #geom_boxplot(outlier.shape = NA) +
      geom_point(aes(color = Transect), alpha = 0.5, size = 2) + 
      geom_line(aes(group = interaction(Transect, metric)), position = position_dodge(width = 0.5))+
      #scale_color_viridis_c(name = "Elevation", option = "viridis") +
      scale_color_manual(name="Transect",
          breaks=c("Aisa","Aran","Sentein","Tourmalet"),
          labels=c("Aisa","Aran","Sentein","Tourmalet"),
          values=c("#e5bd5b50", "#6b739850","#e2815a50", "#876b9650")) +
      facet_wrap(. ~ metric, scales = "free", ncol=4) +
      coord_cartesian(xlim = c(1, NA)) +
      #stat_compare_means(size=2) +
      theme_classic() +
      theme(
        strip.background = element_blank(),
        panel.grid.minor.x = element_line(size = .1, color = "grey"),
        axis.text.x = element_text(angle = 45, hjust = 1)
      ) + 
  ylab("Beta diversity")

#bxp1 + 
  #geom_signif(comparisons = list(c("Aisa", "Aran"), c("Aisa", "Sentein"), c("Aisa", "Tourmalet"),c("Aran", "Sentein"), c("Aran", "Tourmalet"), 
                                # c("Sentein", "Tourmalet")),
             # map_signif_level = TRUE)
```

### Beta diversity per transect (calculated from individuals)

#### Aisa transect

```{r aisa_samples, echo=FALSE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
Aisa_transect<-sample_metadata %>%
    filter(Transect=="Aisa")%>%
    column_to_rownames("EHI_number")

Aisa_transect_nmds<-sample_metadata %>%
    filter(Transect=="Aisa")

# Create a temporary modified version of genome_counts_filt
temp_genome_counts <- transform(genome_counts_filt, row.names = genome_counts_filt$genome)
temp_genome_counts$genome <- NULL
temp_genome_counts_func <- transform(genome_counts_filt_filt, row.names = genome_counts_filt_filt$genome)
temp_genome_counts_func$genome <- NULL

Aisa_counts <- temp_genome_counts[, which(colnames(temp_genome_counts) %in% rownames(Aisa_transect))]
identical(sort(colnames(Aisa_counts)), sort(as.character(rownames(Aisa_transect))))
Aisa_counts_func <- temp_genome_counts_func[, which(colnames(temp_genome_counts_func) %in% rownames(Aisa_transect))]
```

```{r aisa_samples_1, echo=TRUE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
nrow(Aisa_transect)
```

```{r beta_div_aisa, echo=TRUE, results = FALSE, message=FALSE, warning=FALSE, comments=FALSE, eval=FALSE}
beta_div_richness_aisa<-hillpair(data=Aisa_counts, q=0)
beta_div_neutral_aisa<-hillpair(data=Aisa_counts, q=1)
beta_div_phylo_aisa<-hillpair(data=Aisa_counts, q=1, tree=genome_tree)
beta_div_func_aisa<-hillpair(data=Aisa_counts_func, q=1, dist=dist)
```

```{r beta_div_aisa_reg, echo=TRUE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
adonis2(beta_div_richness_aisa$S ~ Elevation+Sampling_point, 
        data = Aisa_transect %>% arrange(match(sample,labels(beta_div_richness_aisa$S))), 
        permutations = 999,
        by="terms") %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_aisa_reg_1, echo=TRUE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
adonis2(beta_div_neutral_aisa$S ~ Elevation+Sampling_point, 
        data = Aisa_transect %>% arrange(match(sample,labels(beta_div_neutral_aisa$S))), 
        permutations = 999,
        by="terms") %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_aisa_reg_2, echo=TRUE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
adonis2(beta_div_phylo_aisa$S ~ Elevation+Sampling_point, 
        data = Aisa_transect %>% arrange(match(sample,labels(beta_div_phylo_aisa$S))), 
        permutations = 999,
        by="terms") %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_aisa_reg_3, echo=TRUE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
adonis2(beta_div_func_aisa$S ~ Elevation+Sampling_point, 
        data = Aisa_transect %>% arrange(match(sample,labels(beta_div_func_aisa$S))), 
        permutations = 999,
        by="terms") %>%
        broom::tidy() %>%
        tt()
```

#### Aran transect

```{r aran_samples, echo=FALSE, results = FALSE, message=FALSE, warning=FALSE, comments=FALSE}
Aran_transect<-sample_metadata %>%
    filter(Transect=="Aran")%>%
    column_to_rownames("EHI_number")

Aran_transect_nmds<-sample_metadata %>%
    filter(Transect=="Aran")

Aran_counts <- temp_genome_counts[, which(colnames(temp_genome_counts) %in% rownames(Aran_transect))]
identical(sort(colnames(Aran_counts)), sort(as.character(rownames(Aran_transect))))
Aran_counts_func <- temp_genome_counts_func[, which(colnames(temp_genome_counts_func) %in% rownames(Aran_transect))]
```

```{r aran_samples_1, echo=FALSE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
nrow(Aran_transect)
```

```{r beta_div_aran, echo=TRUE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE, eval=FALSE}
beta_div_richness_aran<-hillpair(data=Aran_counts, q=0)
beta_div_neutral_aran<-hillpair(data=Aran_counts, q=1)
beta_div_phylo_aran<-hillpair(data=Aran_counts, q=1, tree=genome_tree)
beta_div_func_aran<-hillpair(data=Aran_counts_func, q=1, dist=dist)
```
```{r beta_div_aran_reg, echo=TRUE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
adonis2(beta_div_richness_aran$S ~ Elevation+Sampling_point, 
        data = Aran_transect %>% arrange(match(sample,labels(beta_div_richness_aran$S))), 
        permutations = 999,
        by="terms") %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_aran_reg_1, echo=TRUE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
adonis2(beta_div_neutral_aran$S ~ Elevation+Sampling_point, 
        data = Aran_transect %>% arrange(match(sample,labels(beta_div_neutral_aran$S))), 
        permutations = 999,
        by="terms") %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_aran_reg_2,echo=TRUE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
adonis2(beta_div_phylo_aran$S ~ Elevation+Sampling_point, 
        data = Aran_transect %>% arrange(match(sample,labels(beta_div_phylo_aran$S))), 
        permutations = 999,
        by="terms") %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_aran_reg_3, echo=TRUE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
adonis2(beta_div_func_aran$S ~ Elevation+Sampling_point, 
        data = Aran_transect %>% arrange(match(sample,labels(beta_div_func_aran$S))), 
        permutations = 999,
        by="terms") %>%
        broom::tidy() %>%
        tt()
```


#### Sentein transect

```{r sentein_samples, echo=FALSE, results = FALSE, message=FALSE, warning=FALSE, comments=FALSE}
sentein_transect<-sample_metadata %>%
    filter(Transect=="Sentein")%>%
    column_to_rownames("EHI_number")

sentein_transect_nmds<-sample_metadata %>%
    filter(Transect=="Sentein")

sentein_counts <- temp_genome_counts[, which(colnames(temp_genome_counts) %in% rownames(sentein_transect))]
identical(sort(colnames(sentein_counts)), sort(as.character(rownames(sentein_transect))))
sentein_counts_func <- temp_genome_counts_func[, which(colnames(temp_genome_counts_func) %in% rownames(sentein_transect))]

```

```{r sentein_samples_1, echo=TRUE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
nrow(sentein_transect)
```

```{r beta_div_sentein, echo=TRUE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE, eval=FALSE}
beta_div_richness_sentein<-hillpair(data=sentein_counts, q=0)
beta_div_neutral_sentein<-hillpair(data=sentein_counts, q=1)
beta_div_phylo_sentein<-hillpair(data=sentein_counts, q=1, tree=genome_tree)
beta_div_func_sentein<-hillpair(data=sentein_counts_func, q=1, dist=dist)
```

```{r beta_div_sentein_reg, echo=TRUE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
adonis2(beta_div_richness_sentein$S ~ Elevation+Sampling_point, 
        data = sentein_transect %>% arrange(match(sample,labels(beta_div_richness_sentein$S))), 
        permutations = 999,
        by="terms") %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_sentein_reg_1, echo=TRUE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
adonis2(beta_div_neutral_sentein$S ~ Elevation+Sampling_point, 
        data = sentein_transect %>% arrange(match(sample,labels(beta_div_neutral_sentein$S))), 
        permutations = 999,
        by="terms") %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_sentein_reg_2, echo=TRUE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
adonis2(beta_div_phylo_sentein$S ~ Elevation+Sampling_point, 
        data = sentein_transect %>% arrange(match(sample,labels(beta_div_phylo_sentein$S))), 
        permutations = 999,
        by="terms") %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_sentein_reg_3, echo=TRUE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
adonis2(beta_div_func_sentein$S ~ Elevation+Sampling_point, 
        data = sentein_transect %>% arrange(match(sample,labels(beta_div_func_sentein$S))), 
        permutations = 999,
        by="terms") %>%
        broom::tidy() %>%
        tt()
```

#### Tourmalet transect

```{r tourmalet_samples, echo=FALSE, results = FALSE, message=FALSE, warning=FALSE, comments=FALSE}
tourmalet_transect<-sample_metadata %>%
    filter(Transect=="Tourmalet")%>%
    column_to_rownames("EHI_number")

tourmalet_transect_nmds<-sample_metadata %>%
    filter(Transect=="Tourmalet")

tourmalet_counts <- temp_genome_counts[, which(colnames(temp_genome_counts) %in% rownames(tourmalet_transect))]
identical(sort(colnames(tourmalet_counts)), sort(as.character(rownames(tourmalet_transect))))
tourmalet_counts_func <- temp_genome_counts_func[, which(colnames(temp_genome_counts_func) %in% rownames(tourmalet_transect))]

```

```{r tourmalet_samples_1, echo=TRUE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
nrow(tourmalet_transect)
```

```{r beta_div_tourmalet, echo=TRUE, results = FALSE, message=FALSE, warning=FALSE, comments=FALSE, eval=FALSE}
beta_div_richness_tourmalet<-hillpair(data=tourmalet_counts, q=0)
beta_div_neutral_tourmalet<-hillpair(data=tourmalet_counts, q=1)
beta_div_phylo_tourmalet<-hillpair(data=tourmalet_counts, q=1, tree=genome_tree)
beta_div_func_tourmalet<-hillpair(data=tourmalet_counts_func, q=1, dist=dist)
```

```{r beta_div_tourmalet_reg, echo=TRUE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
adonis2(beta_div_richness_tourmalet$S ~ Elevation+Sampling_point, 
        data = tourmalet_transect %>% arrange(match(sample,labels(beta_div_richness_tourmalet$S))), 
        permutations = 999,
        by="terms") %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_tourmalet_reg_1, echo=TRUE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
adonis2(beta_div_neutral_tourmalet$S ~ Elevation+Sampling_point, 
        data = tourmalet_transect %>% arrange(match(sample,labels(beta_div_neutral_tourmalet$S))), 
        permutations = 999,
        by="terms") %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_tourmalet_reg_2, echo=TRUE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
adonis2(beta_div_phylo_tourmalet$S ~ Elevation+Sampling_point, 
        data = tourmalet_transect %>% arrange(match(sample,labels(beta_div_phylo_tourmalet$S))), 
        permutations = 999,
        by="terms") %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_tourmalet_reg_3, echo=TRUE, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
adonis2(beta_div_func_tourmalet$S ~ Elevation+Sampling_point, 
        data = tourmalet_transect %>% arrange(match(sample,labels(beta_div_func_tourmalet$S))), 
        permutations = 999,
        by="terms") %>%
        broom::tidy() %>%
        tt()
```

### Beta diversity per transect (calculated from elevation chunks)

```{r transect_data, eval=FALSE}
Aisa_transect<-sample_metadata %>%
    filter(Transect=="Aisa")%>%
    column_to_rownames("EHI_number")

Aran_transect<-sample_metadata %>%
    filter(Transect=="Aran")%>%
    column_to_rownames("EHI_number")

sentein_transect<-sample_metadata %>%
    filter(Transect=="Sentein")%>%
    column_to_rownames("EHI_number")

tourmalet_transect<-sample_metadata %>%
    filter(Transect=="Tourmalet")%>%
    column_to_rownames("EHI_number")
```


#### Aisa

```{r aisa_transect_data_A, eval=FALSE}
Aisa_transect_A<-Aisa_transect %>%
  filter(Elevation=="1250" | Elevation=="1450")

Aisa_counts_A <- temp_genome_counts[, which(colnames(temp_genome_counts) %in% rownames(Aisa_transect_A))]
identical(sort(colnames(Aisa_counts_A)), sort(as.character(rownames(Aisa_transect_A))))
Aisa_counts_func_A <- temp_genome_counts_func[, which(colnames(temp_genome_counts_func) %in% rownames(Aisa_transect_A))]
```

```{r aisa_transect_beta_div_A, eval=FALSE}
beta_div_richness_aisa_a<-hillpair(data=Aisa_counts_A, q=0)
beta_div_neutral_aisa_a<-hillpair(data=Aisa_counts_A, q=1)
beta_div_phylo_aisa_a<-hillpair(data=Aisa_counts_A, q=1, tree=genome_tree)
beta_div_func_aisa_a<-hillpair(data=Aisa_counts_func_A, q=1, dist=dist)
```

```{r aisa_transect_pairwise_A, eval=TRUE, results=TRUE}
pairwise_richness<-pairwise.adonis(beta_div_richness_aisa_a$S, Aisa_transect_A$Elevation, perm = 999)
pairwise_richness%>%
  tt()
pairwise_neutral<-pairwise.adonis(beta_div_neutral_aisa_a$S, Aisa_transect_A$Elevation, perm = 999)
pairwise_neutral%>%
  tt()
pairwise_phylo<-pairwise.adonis(beta_div_phylo_aisa_a$S, Aisa_transect_A$Elevation, perm = 999)
pairwise_phylo%>%
  tt()
pairwise_function<-pairwise.adonis(beta_div_func_aisa_a$S, Aisa_transect_A$Elevation, perm = 999)
pairwise_function%>%
  tt()
```

```{r aisa_transect_data_B, eval=FALSE}
Aisa_transect_B<-Aisa_transect %>%
  filter(Elevation=="1450" | Elevation=="1650")

Aisa_counts_B <- temp_genome_counts[, which(colnames(temp_genome_counts) %in% rownames(Aisa_transect_B))]
identical(sort(colnames(Aisa_counts_B)), sort(as.character(rownames(Aisa_transect_B))))
Aisa_counts_func_B <- temp_genome_counts_func[, which(colnames(temp_genome_counts_func) %in% rownames(Aisa_transect_B))]
```

```{r aisa_transect_beta_div_B, eval=FALSE}
beta_div_richness_aisa_b<-hillpair(data=Aisa_counts_B, q=0)
beta_div_neutral_aisa_b<-hillpair(data=Aisa_counts_B, q=1)
beta_div_phylo_aisa_b<-hillpair(data=Aisa_counts_B, q=1, tree=genome_tree)
beta_div_func_aisa_b<-hillpair(data=Aisa_counts_func_B, q=1, dist=dist)
```

```{r aisa_transect_pairwise_B, eval=TRUE, results=TRUE}
pairwise_richness<-pairwise.adonis(beta_div_richness_aisa_b$S, Aisa_transect_B$Elevation, perm = 999)
pairwise_richness%>%
  tt()
pairwise_neutral<-pairwise.adonis(beta_div_neutral_aisa_b$S, Aisa_transect_B$Elevation, perm = 999)
pairwise_neutral%>%
  tt()
pairwise_phylo<-pairwise.adonis(beta_div_phylo_aisa_b$S, Aisa_transect_B$Elevation, perm = 999)
pairwise_phylo%>%
  tt()
pairwise_function<-pairwise.adonis(beta_div_func_aisa_b$S, Aisa_transect_B$Elevation, perm = 999)
pairwise_function%>%
  tt()
```

```{r aisa_transect_data_C, eval=FALSE}
Aisa_transect_C<-Aisa_transect %>%
  filter(Elevation=="1650" | Elevation=="1850")

Aisa_counts_C <- temp_genome_counts[, which(colnames(temp_genome_counts) %in% rownames(Aisa_transect_C))]
identical(sort(colnames(Aisa_counts_C)), sort(as.character(rownames(Aisa_transect_C))))
Aisa_counts_func_C <- temp_genome_counts_func[, which(colnames(temp_genome_counts_func) %in% rownames(Aisa_transect_C))]
```

```{r aisa_transect_beta_div_C, eval=FALSE}
beta_div_richness_aisa_c<-hillpair(data=Aisa_counts_C, q=0)
beta_div_neutral_aisa_c<-hillpair(data=Aisa_counts_C, q=1)
beta_div_phylo_aisa_c<-hillpair(data=Aisa_counts_C, q=1, tree=genome_tree)
beta_div_func_aisa_c<-hillpair(data=Aisa_counts_func_C, q=1, dist=dist)
```

```{r aisa_transect_pairwise_C, eval=TRUE, results=TRUE}
pairwise_richness<-pairwise.adonis(beta_div_richness_aisa_c$S, Aisa_transect_C$Elevation, perm = 999)
pairwise_richness%>%
  tt()
pairwise_neutral<-pairwise.adonis(beta_div_neutral_aisa_c$S, Aisa_transect_C$Elevation, perm = 999)
pairwise_neutral%>%
  tt()
pairwise_phylo<-pairwise.adonis(beta_div_phylo_aisa_c$S, Aisa_transect_C$Elevation, perm = 999)
pairwise_phylo%>%
  tt()
pairwise_function<-pairwise.adonis(beta_div_func_aisa_c$S, Aisa_transect_C$Elevation, perm = 999)
pairwise_function%>%
  tt()
```

#### Aran

```{r aran_transect_data_A, eval=FALSE}
Aran_transect_A<-Aran_transect %>%
  filter(Elevation=="1000" | Elevation=="1080")

Aran_counts_A <- temp_genome_counts[, which(colnames(temp_genome_counts) %in% rownames(Aran_transect_A))]
identical(sort(colnames(Aran_counts_A)), sort(as.character(rownames(Aran_transect_A))))
Aran_counts_func_A <- temp_genome_counts_func[, which(colnames(temp_genome_counts_func) %in% rownames(Aran_transect_A))]
```

```{r aran_transect_beta_div_A, eval=FALSE}
beta_div_richness_aran_a<-hillpair(data=Aran_counts_A, q=0)
beta_div_neutral_aran_a<-hillpair(data=Aran_counts_A, q=1)
beta_div_phylo_aran_a<-hillpair(data=Aran_counts_A, q=1, tree=genome_tree)
beta_div_func_aran_a<-hillpair(data=Aran_counts_func_A, q=1, dist=dist)
```

```{r aran_transect_pairwise_A, eval=TRUE, results=TRUE}
pairwise_richness<-pairwise.adonis(beta_div_richness_aran_a$S, Aran_transect_A$Elevation, perm = 999)
pairwise_richness%>%
  tt()
pairwise_neutral<-pairwise.adonis(beta_div_neutral_aran_a$S, Aran_transect_A$Elevation, perm = 999)
pairwise_neutral%>%
  tt()
pairwise_phylo<-pairwise.adonis(beta_div_phylo_aran_a$S, Aran_transect_A$Elevation, perm = 999)
pairwise_phylo%>%
  tt()
pairwise_function<-pairwise.adonis(beta_div_func_aran_a$S, Aran_transect_A$Elevation, perm = 999)
pairwise_function%>%
  tt()
```

```{r aran_transect_data_B, eval=FALSE}
Aran_transect_B<-Aran_transect %>%
  filter(Elevation=="1080" | Elevation=="1340")

Aran_counts_B <- temp_genome_counts[, which(colnames(temp_genome_counts) %in% rownames(Aran_transect_B))]
identical(sort(colnames(Aran_counts_B)), sort(as.character(rownames(Aran_transect_B))))
Aran_counts_func_B <- temp_genome_counts_func[, which(colnames(temp_genome_counts_func) %in% rownames(Aran_transect_B))]
```

```{r aran_transect_beta_div_B, eval=FALSE}
beta_div_richness_aran_b<-hillpair(data=Aran_counts_B, q=0)
beta_div_neutral_aran_b<-hillpair(data=Aran_counts_B, q=1)
beta_div_phylo_aran_b<-hillpair(data=Aran_counts_B, q=1, tree=genome_tree)
beta_div_func_aran_b<-hillpair(data=Aran_counts_func_B, q=1, dist=dist)
```

```{r aran_transect_pairwise_B, eval=TRUE, results=TRUE}
pairwise_richness<-pairwise.adonis(beta_div_richness_aran_b$S, Aran_transect_B$Elevation, perm = 999)
pairwise_richness%>%
  tt()
pairwise_neutral<-pairwise.adonis(beta_div_neutral_aran_b$S, Aran_transect_B$Elevation, perm = 999)
pairwise_neutral%>%
  tt()
pairwise_phylo<-pairwise.adonis(beta_div_phylo_aran_b$S, Aran_transect_B$Elevation, perm = 999)
pairwise_phylo%>%
  tt()
pairwise_function<-pairwise.adonis(beta_div_func_aran_b$S, Aran_transect_B$Elevation, perm = 999)
pairwise_function%>%
  tt()
```

```{r aran_transect_data_C, eval=FALSE}
Aran_transect_C<-Aran_transect %>%
  filter(Elevation=="1340" | Elevation=="1500")

Aran_counts_C <- temp_genome_counts[, which(colnames(temp_genome_counts) %in% rownames(Aran_transect_C))]
identical(sort(colnames(Aran_counts_C)), sort(as.character(rownames(Aran_transect_C))))
Aran_counts_func_C <- temp_genome_counts_func[, which(colnames(temp_genome_counts_func) %in% rownames(Aran_transect_C))]
```

```{r aran_transect_beta_div_C, eval=FALSE}
beta_div_richness_aran_c<-hillpair(data=Aran_counts_C, q=0)
beta_div_neutral_aran_c<-hillpair(data=Aran_counts_C, q=1)
beta_div_phylo_aran_c<-hillpair(data=Aran_counts_C, q=1, tree=genome_tree)
beta_div_func_aran_c<-hillpair(data=Aran_counts_func_C, q=1, dist=dist)
```

```{r aran_transect_pairwise_C, eval=TRUE, results=TRUE}
pairwise_richness<-pairwise.adonis(beta_div_richness_aran_c$S, Aran_transect_C$Elevation, perm = 999)
pairwise_richness%>%
  tt()
pairwise_neutral<-pairwise.adonis(beta_div_neutral_aran_c$S, Aran_transect_C$Elevation, perm = 999)
pairwise_neutral%>%
  tt()
pairwise_phylo<-pairwise.adonis(beta_div_phylo_aran_c$S, Aran_transect_C$Elevation, perm = 999)
pairwise_phylo%>%
  tt()
pairwise_function<-pairwise.adonis(beta_div_func_aran_c$S, Aran_transect_C$Elevation, perm = 999)
pairwise_function%>%
  tt()
```

```{r aran_transect_data_D, eval=FALSE}
Aran_transect_D<-Aran_transect %>%
  filter(Elevation=="1500" | Elevation=="1650")

Aran_counts_D <- temp_genome_counts[, which(colnames(temp_genome_counts) %in% rownames(Aran_transect_D))]
identical(sort(colnames(Aran_counts_D)), sort(as.character(rownames(Aran_transect_D))))
Aran_counts_func_D <- temp_genome_counts_func[, which(colnames(temp_genome_counts_func) %in% rownames(Aran_transect_D))]
```

```{r aran_transect_beta_div_D, eval=FALSE}
beta_div_richness_aran_d<-hillpair(data=Aran_counts_D, q=0)
beta_div_neutral_aran_d<-hillpair(data=Aran_counts_D, q=1)
beta_div_phylo_aran_d<-hillpair(data=Aran_counts_D, q=1, tree=genome_tree)
beta_div_func_aran_d<-hillpair(data=Aran_counts_func_D, q=1, dist=dist)
```

```{r aran_transect_pairwise_D, eval=TRUE, results=TRUE}
pairwise_richness<-pairwise.adonis(beta_div_richness_aran_d$S, Aran_transect_D$Elevation, perm = 999)
pairwise_richness%>%
  tt()
pairwise_neutral<-pairwise.adonis(beta_div_neutral_aran_d$S, Aran_transect_D$Elevation, perm = 999)
pairwise_neutral%>%
  tt()
pairwise_phylo<-pairwise.adonis(beta_div_phylo_aran_d$S, Aran_transect_D$Elevation, perm = 999)
pairwise_phylo%>%
  tt()
pairwise_function<-pairwise.adonis(beta_div_func_aran_d$S, Aran_transect_D$Elevation, perm = 999)
pairwise_function%>%
  tt()
```

```{r aran_transect_data_E, eval=FALSE}
Aran_transect_E<-Aran_transect %>%
  filter(Elevation=="1650" | Elevation=="1850")

Aran_counts_E <- temp_genome_counts[, which(colnames(temp_genome_counts) %in% rownames(Aran_transect_E))]
identical(sort(colnames(Aran_counts_E)), sort(as.character(rownames(Aran_transect_E))))
Aran_counts_func_E <- temp_genome_counts_func[, which(colnames(temp_genome_counts_func) %in% rownames(Aran_transect_E))]
```

```{r aran_transect_beta_div_E, eval=FALSE}
beta_div_richness_aran_e<-hillpair(data=Aran_counts_E, q=0)
beta_div_neutral_aran_e<-hillpair(data=Aran_counts_E, q=1)
beta_div_phylo_aran_e<-hillpair(data=Aran_counts_E, q=1, tree=genome_tree)
beta_div_func_aran_e<-hillpair(data=Aran_counts_func_E, q=1, dist=dist)
```

```{r aran_transect_pairwise_E, eval=TRUE, results=TRUE}
pairwise_richness<-pairwise.adonis(beta_div_richness_aran_e$S, Aran_transect_E$Elevation, perm = 999)
pairwise_richness%>%
  tt()
pairwise_neutral<-pairwise.adonis(beta_div_neutral_aran_e$S, Aran_transect_E$Elevation, perm = 999)
pairwise_neutral%>%
  tt()
pairwise_phylo<-pairwise.adonis(beta_div_phylo_aran_e$S, Aran_transect_E$Elevation, perm = 999)
pairwise_phylo%>%
  tt()
pairwise_function<-pairwise.adonis(beta_div_func_aran_e$S, Aran_transect_E$Elevation, perm = 999)
pairwise_function%>%
  tt()
```

#### Sentein

```{r sentein_transect_data_A, eval=FALSE}
sentein_transect_A<-sentein_transect %>%
  filter(Elevation=="941" | Elevation=="1260")

sentein_counts_A <- temp_genome_counts[, which(colnames(temp_genome_counts) %in% rownames(sentein_transect_A))]
identical(sort(colnames(sentein_counts_A)), sort(as.character(rownames(sentein_transect_A))))
sentein_counts_func_A <- temp_genome_counts_func[, which(colnames(temp_genome_counts_func) %in% rownames(sentein_transect_A))]
```

```{r sentein_transect_beta_div_A, eval=FALSE}
beta_div_richness_sentein_a<-hillpair(data=sentein_counts_A, q=0)
beta_div_neutral_sentein_a<-hillpair(data=sentein_counts_A, q=1)
beta_div_phylo_sentein_a<-hillpair(data=sentein_counts_A, q=1, tree=genome_tree)
beta_div_func_sentein_a<-hillpair(data=sentein_counts_func_A, q=1, dist=dist)
```

```{r sentein_transect_pairwise_A, eval=TRUE, results=TRUE}
pairwise_richness<-pairwise.adonis(beta_div_richness_sentein_a$S, sentein_transect_A$Elevation, perm = 999)
pairwise_richness%>%
  tt()
pairwise_neutral<-pairwise.adonis(beta_div_neutral_sentein_a$S, sentein_transect_A$Elevation, perm = 999)
pairwise_neutral%>%
  tt()
pairwise_phylo<-pairwise.adonis(beta_div_phylo_sentein_a$S, sentein_transect_A$Elevation, perm = 999)
pairwise_phylo%>%
  tt()
pairwise_function<-pairwise.adonis(beta_div_func_sentein_a$S, sentein_transect_A$Elevation, perm = 999)
pairwise_function%>%
  tt()
```

```{r sentein_transect_data_B, eval=FALSE}
sentein_transect_B<-sentein_transect %>%
  filter(Elevation=="1260" | Elevation=="1628")

sentein_counts_B <- temp_genome_counts[, which(colnames(temp_genome_counts) %in% rownames(sentein_transect_B))]
identical(sort(colnames(sentein_counts_B)), sort(as.character(rownames(sentein_transect_B))))
sentein_counts_func_B <- temp_genome_counts_func[, which(colnames(temp_genome_counts_func) %in% rownames(sentein_transect_B))]
```

```{r sentein_transect_beta_div_B, eval=FALSE}
beta_div_richness_sentein_b<-hillpair(data=sentein_counts_B, q=0)
beta_div_neutral_sentein_b<-hillpair(data=sentein_counts_B, q=1)
beta_div_phylo_sentein_b<-hillpair(data=sentein_counts_B, q=1, tree=genome_tree)
beta_div_func_sentein_b<-hillpair(data=sentein_counts_func_B, q=1, dist=dist)
```

```{r sentein_transect_pairwise_B, eval=TRUE, results=TRUE}
pairwise_richness<-pairwise.adonis(beta_div_richness_sentein_b$S, sentein_transect_B$Elevation, perm = 999)
pairwise_richness%>%
  tt()
pairwise_neutral<-pairwise.adonis(beta_div_neutral_sentein_b$S, sentein_transect_B$Elevation, perm = 999)
pairwise_neutral%>%
  tt()
pairwise_phylo<-pairwise.adonis(beta_div_phylo_sentein_b$S, sentein_transect_B$Elevation, perm = 999)
pairwise_phylo%>%
  tt()
pairwise_function<-pairwise.adonis(beta_div_func_sentein_b$S, sentein_transect_B$Elevation, perm = 999)
pairwise_function%>%
  tt()
```

```{r sentein_transect_data_C, eval=FALSE}
sentein_transect_C<-sentein_transect %>%
  filter(Elevation=="1628" | Elevation=="1873")

sentein_counts_C <- temp_genome_counts[, which(colnames(temp_genome_counts) %in% rownames(sentein_transect_C))]
identical(sort(colnames(sentein_counts_C)), sort(as.character(rownames(sentein_transect_C))))
sentein_counts_func_C <- temp_genome_counts_func[, which(colnames(temp_genome_counts_func) %in% rownames(sentein_transect_C))]
```

```{r sentein_transect_beta_div_C, eval=FALSE}
beta_div_richness_sentein_c<-hillpair(data=sentein_counts_C, q=0)
beta_div_neutral_sentein_c<-hillpair(data=sentein_counts_C, q=1)
beta_div_phylo_sentein_c<-hillpair(data=sentein_counts_C, q=1, tree=genome_tree)
beta_div_func_sentein_c<-hillpair(data=sentein_counts_func_C, q=1, dist=dist)
```

```{r sentein_transect_pairwise_C, eval=TRUE, results=TRUE}
pairwise_richness<-pairwise.adonis(beta_div_richness_sentein_c$S, sentein_transect_C$Elevation, perm = 999)
pairwise_richness%>%
  tt()
pairwise_neutral<-pairwise.adonis(beta_div_neutral_sentein_c$S, sentein_transect_C$Elevation, perm = 999)
pairwise_neutral%>%
  tt()
pairwise_phylo<-pairwise.adonis(beta_div_phylo_sentein_c$S, sentein_transect_C$Elevation, perm = 999)
pairwise_phylo%>%
  tt()
pairwise_function<-pairwise.adonis(beta_div_func_sentein_c$S, sentein_transect_C$Elevation, perm = 999)
pairwise_function%>%
  tt()
```

#### Tourmalet

```{r tourmalet_transect_data_A, eval=FALSE}
tourmalet_transect_A<-tourmalet_transect %>%
  filter(Elevation=="953" | Elevation=="1255")

tourmalet_counts_A <- temp_genome_counts[, which(colnames(temp_genome_counts) %in% rownames(tourmalet_transect_A))]
identical(sort(colnames(tourmalet_counts_A)), sort(as.character(rownames(tourmalet_transect_A))))
tourmalet_counts_func_A <- temp_genome_counts_func[, which(colnames(temp_genome_counts_func) %in% rownames(tourmalet_transect_A))]
```

```{r tourmalet_transect_beta_div_A, eval=FALSE}
beta_div_richness_tourmalet_a<-hillpair(data=tourmalet_counts_A, q=0)
beta_div_neutral_tourmalet_a<-hillpair(data=tourmalet_counts_A, q=1)
beta_div_phylo_tourmalet_a<-hillpair(data=tourmalet_counts_A, q=1, tree=genome_tree)
beta_div_func_tourmalet_a<-hillpair(data=tourmalet_counts_func_A, q=1, dist=dist)
```

```{r tourmalet_transect_pairwise_A, eval=TRUE, results=TRUE}
pairwise_richness<-pairwise.adonis(beta_div_richness_tourmalet_a$S, tourmalet_transect_A$Elevation, perm = 999)
pairwise_richness%>%
  tt()
pairwise_neutral<-pairwise.adonis(beta_div_neutral_tourmalet_a$S, tourmalet_transect_A$Elevation, perm = 999)
pairwise_neutral%>%
  tt()
pairwise_phylo<-pairwise.adonis(beta_div_phylo_tourmalet_a$S, tourmalet_transect_A$Elevation, perm = 999)
pairwise_phylo%>%
  tt()
pairwise_function<-pairwise.adonis(beta_div_func_tourmalet_a$S, tourmalet_transect_A$Elevation, perm = 999)
pairwise_function%>%
  tt()
```

```{r tourmalet_transect_data_B, eval=FALSE}
tourmalet_transect_B<-tourmalet_transect %>%
  filter(Elevation=="1255" | Elevation=="1561")

tourmalet_counts_B <- temp_genome_counts[, which(colnames(temp_genome_counts) %in% rownames(tourmalet_transect_B))]
identical(sort(colnames(tourmalet_counts_B)), sort(as.character(rownames(tourmalet_transect_B))))
tourmalet_counts_func_B <- temp_genome_counts_func[, which(colnames(temp_genome_counts_func) %in% rownames(tourmalet_transect_B))]
```

```{r tourmalet_transect_beta_div_B, eval=FALSE}
beta_div_richness_tourmalet_b<-hillpair(data=tourmalet_counts_B, q=0)
beta_div_neutral_tourmalet_b<-hillpair(data=tourmalet_counts_B, q=1)
beta_div_phylo_tourmalet_b<-hillpair(data=tourmalet_counts_B, q=1, tree=genome_tree)
beta_div_func_tourmalet_b<-hillpair(data=tourmalet_counts_func_B, q=1, dist=dist)
```

```{r tourmalet_transect_pairwise_B, eval=TRUE, results=TRUE}
pairwise_richness<-pairwise.adonis(beta_div_richness_tourmalet_b$S, tourmalet_transect_B$Elevation, perm = 999)
pairwise_richness%>%
  tt()
pairwise_neutral<-pairwise.adonis(beta_div_neutral_tourmalet_b$S, tourmalet_transect_B$Elevation, perm = 999)
pairwise_neutral%>%
  tt()
pairwise_phylo<-pairwise.adonis(beta_div_phylo_tourmalet_b$S, tourmalet_transect_B$Elevation, perm = 999)
pairwise_phylo%>%
  tt()
pairwise_function<-pairwise.adonis(beta_div_func_tourmalet_b$S, tourmalet_transect_B$Elevation, perm = 999)
pairwise_function%>%
  tt()
```

```{r tourmalet_transect_data_C, eval=FALSE}
tourmalet_transect_C<-tourmalet_transect %>%
  filter(Elevation=="1561" | Elevation=="1797")

tourmalet_counts_C <- temp_genome_counts[, which(colnames(temp_genome_counts) %in% rownames(tourmalet_transect_C))]
identical(sort(colnames(tourmalet_counts_C)), sort(as.character(rownames(tourmalet_transect_C))))
tourmalet_counts_func_C <- temp_genome_counts_func[, which(colnames(temp_genome_counts_func) %in% rownames(tourmalet_transect_C))]
```

```{r tourmalet_transect_beta_div_C, eval=FALSE}
beta_div_richness_tourmalet_c<-hillpair(data=tourmalet_counts_C, q=0)
beta_div_neutral_tourmalet_c<-hillpair(data=tourmalet_counts_C, q=1)
beta_div_phylo_tourmalet_c<-hillpair(data=tourmalet_counts_C, q=1, tree=genome_tree)
beta_div_func_tourmalet_c<-hillpair(data=tourmalet_counts_func_C, q=1, dist=dist)
```

```{r tourmalet_transect_pairwise_C, eval=TRUE, results=TRUE}
pairwise_richness<-pairwise.adonis(beta_div_richness_tourmalet_c$S, tourmalet_transect_C$Elevation, perm = 999)
pairwise_richness%>%
  tt()
pairwise_neutral<-pairwise.adonis(beta_div_neutral_tourmalet_c$S, tourmalet_transect_C$Elevation, perm = 999)
pairwise_neutral%>%
  tt()
pairwise_phylo<-pairwise.adonis(beta_div_phylo_tourmalet_c$S, tourmalet_transect_C$Elevation, perm = 999)
pairwise_phylo%>%
  tt()
pairwise_function<-pairwise.adonis(beta_div_func_tourmalet_c$S, tourmalet_transect_C$Elevation, perm = 999)
pairwise_function%>%
  tt()
```

```{r tourmalet_transect_data_D, eval=FALSE}
tourmalet_transect_D<-tourmalet_transect %>%
  filter(Elevation=="1797" | Elevation=="2065"| Elevation=="2134")

tourmalet_counts_D <- temp_genome_counts[, which(colnames(temp_genome_counts) %in% rownames(tourmalet_transect_D))]
identical(sort(colnames(tourmalet_counts_D)), sort(as.character(rownames(tourmalet_transect_D))))
tourmalet_counts_func_D <- temp_genome_counts_func[, which(colnames(temp_genome_counts_func) %in% rownames(tourmalet_transect_D))]
```

```{r tourmalet_transect_beta_div_D, eval=FALSE}
beta_div_richness_tourmalet_d<-hillpair(data=tourmalet_counts_D, q=0)
beta_div_neutral_tourmalet_d<-hillpair(data=tourmalet_counts_D, q=1)
beta_div_phylo_tourmalet_d<-hillpair(data=tourmalet_counts_D, q=1, tree=genome_tree)
beta_div_func_tourmalet_d<-hillpair(data=tourmalet_counts_func_D, q=1, dist=dist)
```

```{r tourmalet_transect_pairwise_D, eval=TRUE, results=TRUE, warning=FALSE, comment=FALSE}
pairwise_richness<-pairwise.adonis(beta_div_richness_tourmalet_d$S, tourmalet_transect_D$Elevation, perm = 999)
pairwise_richness%>%
  tt()
pairwise_neutral<-pairwise.adonis(beta_div_neutral_tourmalet_d$S, tourmalet_transect_D$Elevation, perm = 999)
pairwise_neutral%>%
  tt()
pairwise_phylo<-pairwise.adonis(beta_div_phylo_tourmalet_d$S, tourmalet_transect_D$Elevation, perm = 999)
pairwise_phylo%>%
  tt()
pairwise_function<-pairwise.adonis(beta_div_func_tourmalet_d$S, tourmalet_transect_D$Elevation, perm = 999)
pairwise_function%>%
  tt()
```

```{r tourmalet_transect_data_E, eval=FALSE}
tourmalet_transect_E<-tourmalet_transect %>%
  filter(Elevation=="2065"| Elevation=="2134"| Elevation=="2306")

tourmalet_counts_E <- temp_genome_counts[, which(colnames(temp_genome_counts) %in% rownames(tourmalet_transect_E))]
identical(sort(colnames(tourmalet_counts_E)), sort(as.character(rownames(tourmalet_transect_E))))
tourmalet_counts_func_E <- temp_genome_counts_func[, which(colnames(temp_genome_counts_func) %in% rownames(tourmalet_transect_E))]
```

```{r tourmalet_transect_beta_div_E, eval=FALSE}
beta_div_richness_tourmalet_e<-hillpair(data=tourmalet_counts_E, q=0)
beta_div_neutral_tourmalet_e<-hillpair(data=tourmalet_counts_E, q=1)
beta_div_phylo_tourmalet_e<-hillpair(data=tourmalet_counts_E, q=1, tree=genome_tree)
beta_div_func_tourmalet_e<-hillpair(data=tourmalet_counts_func_E, q=1, dist=dist)
```

```{r tourmalet_transect_pairwise_E, eval=TRUE, results=TRUE}
pairwise_richness<-pairwise.adonis(beta_div_richness_tourmalet_e$S, tourmalet_transect_E$Elevation, perm = 999)
pairwise_richness%>%
  tt()
pairwise_neutral<-pairwise.adonis(beta_div_neutral_tourmalet_e$S, tourmalet_transect_E$Elevation, perm = 999)
pairwise_neutral%>%
  tt()
pairwise_phylo<-pairwise.adonis(beta_div_phylo_tourmalet_e$S, tourmalet_transect_E$Elevation, perm = 999)
pairwise_phylo%>%
  tt()
pairwise_function<-pairwise.adonis(beta_div_func_tourmalet_e$S, tourmalet_transect_E$Elevation, perm = 999)
pairwise_function%>%
  tt()
```


### Beta diversity per transect (calculated from population)
#### Aisa

```{r aisa_mean, eval=FALSE}

Aisa_transect<-tibble::rownames_to_column(Aisa_transect, var = "sample")
Aisa_counts<-tibble::rownames_to_column(Aisa_counts, var = "genome")


richness_beta_aisa <- Aisa_counts %>%
  pivot_longer(!genome,names_to="sample",values_to = "counts") %>%
  left_join(Aisa_transect, by = join_by(sample == sample)) %>%
  mutate(Elevation=factor(Elevation)) %>% 
  group_by(Elevation) %>% 
  group_split() %>% 
  map_dbl(., ~ .x %>%
            select(genome, sample, counts) %>%
            pivot_wider(names_from = sample, values_from = counts) %>%
            column_to_rownames(var = "genome") %>%
            tss() %>%
            as.data.frame() %>%
            hilldiss(data=., metric="S", q = 0)
  )

richness_beta_aisa <- Aisa_counts %>%
  pivot_longer(!genome, names_to = "sample", values_to = "counts") %>%
  left_join(Aisa_transect, by = join_by(sample == sample)) %>%
  mutate(Elevation = factor(Elevation)) %>%
  group_by(Elevation) %>%
  group_split() %>%
  map(., ~ .x %>%
        select(genome, sample, counts) %>%
        pivot_wider(names_from = sample, values_from = counts) %>%
        column_to_rownames(var = "genome") %>%
        tss() %>%
        as.data.frame() %>%
        hilldiss(data = ., metric = "S", q = 0)) %>%
  do.call(cbind, .)


names(richness_beta_aisa)  <- unique(Aisa_transect$Elevation) %>% sort()

neutral_beta_aisa <- Aisa_counts %>%
  pivot_longer(!genome,names_to="sample",values_to = "counts") %>%
  left_join(Aisa_transect, by = join_by(sample == sample)) %>%
  mutate(Elevation=factor(Elevation)) %>% 
  group_by(Elevation) %>% 
  group_split() %>% 
  map_dbl(., ~ .x %>%
            select(genome, sample, counts) %>%
            pivot_wider(names_from = sample, values_from = counts) %>%
            column_to_rownames(var = "genome") %>%
            tss() %>%
            as.data.frame() %>%
            hilldiss(data=., metric="S", q = 1)
  )

names(neutral_beta_aisa)  <- unique(Aisa_transect$Elevation) %>% sort()

phylo_beta_aisa <- Aisa_counts %>%
  pivot_longer(!genome,names_to="sample",values_to = "counts") %>%
  left_join(Aisa_transect, by = join_by(sample == sample)) %>%
  mutate(Elevation=factor(Elevation)) %>% 
  group_by(Elevation) %>% 
  group_split() %>% 
  map_dbl(., ~ .x %>%
            select(genome, sample, counts) %>%
            pivot_wider(names_from = sample, values_from = counts) %>%
            column_to_rownames(var = "genome") %>%
            tss() %>%
            as.data.frame() %>%
            hilldiss(data=., metric="S", q = 1, tree=genome_tree)
  )

names(phylo_beta_aisa)  <- unique(Aisa_transect$Elevation) %>% sort()

Aisa_counts_func<-tibble::rownames_to_column(Aisa_counts_func, var = "genome")

func_beta_aisa <- Aisa_counts_func %>%
  pivot_longer(!genome,names_to="sample",values_to = "counts") %>%
  left_join(Aisa_transect, by = join_by(sample == sample)) %>%
  mutate(Elevation=factor(Elevation)) %>% 
  group_by(Elevation) %>% 
  group_split() %>% 
  map_dbl(., ~ .x %>%
            select(genome, sample, counts) %>%
            pivot_wider(names_from = sample, values_from = counts) %>%
            column_to_rownames(var = "genome") %>%
            tss() %>%
            as.data.frame() %>%
            hilldiss(data=., metric="S", q = 1, dist=dist)
  )

names(func_beta_aisa)  <- unique(Aisa_transect$Elevation) %>% sort()
```


```{r aisa_mean_1, eval=TRUE, results=TRUE}
# Merge all metrics
beta_div_aisa <- bind_rows(richness_beta_aisa,neutral_beta_aisa,phylo_beta_aisa,func_beta_aisa) %>%
  t() %>% 
  as.data.frame() %>% 
  rownames_to_column(var="Elevation") %>% 
  as_tibble() %>% 
  rename("richness"=V1,"neutral"=V2,"phylogenetic"=V3,"functional"=V4)
```


#### Aran

```{r aran_mean, eval=FALSE}
aran_transect<-tibble::rownames_to_column(Aran_transect, var = "sample")
aran_counts<-tibble::rownames_to_column(Aran_counts, var = "genome")

richness_beta_aran <- aran_counts %>%
  pivot_longer(!genome,names_to="sample",values_to = "counts") %>%
  left_join(aran_transect, by = join_by(sample == sample)) %>%
  mutate(Elevation=factor(Elevation)) %>% 
  group_by(Elevation) %>% 
  group_split() %>% 
  map_dbl(., ~ .x %>%
            select(genome, sample, counts) %>%
            pivot_wider(names_from = sample, values_from = counts) %>%
            column_to_rownames(var = "genome") %>%
            tss() %>%
            as.data.frame() %>%
            hilldiss(data=., metric="S", q = 0)
  )

names(richness_beta_aran)  <- unique(aran_transect$Elevation) %>% sort()

neutral_beta_aran <- aran_counts %>%
  pivot_longer(!genome,names_to="sample",values_to = "counts") %>%
  left_join(aran_transect, by = join_by(sample == sample)) %>%
  mutate(Elevation=factor(Elevation)) %>% 
  group_by(Elevation) %>% 
  group_split() %>% 
  map_dbl(., ~ .x %>%
            select(genome, sample, counts) %>%
            pivot_wider(names_from = sample, values_from = counts) %>%
            column_to_rownames(var = "genome") %>%
            tss() %>%
            as.data.frame() %>%
            hilldiss(data=., metric="S", q = 1)
  )

names(neutral_beta_aran)  <- unique(aran_transect$Elevation) %>% sort()

phylo_beta_aran <- aran_counts %>%
  pivot_longer(!genome,names_to="sample",values_to = "counts") %>%
  left_join(aran_transect, by = join_by(sample == sample)) %>%
  mutate(Elevation=factor(Elevation)) %>% 
  group_by(Elevation) %>% 
  group_split() %>% 
  map_dbl(., ~ .x %>%
            select(genome, sample, counts) %>%
            pivot_wider(names_from = sample, values_from = counts) %>%
            column_to_rownames(var = "genome") %>%
            tss() %>%
            as.data.frame() %>%
            hilldiss(data=., metric="S", q = 1, tree=genome_tree)
  )

names(phylo_beta_aran)  <- unique(aran_transect$Elevation) %>% sort()

aran_counts_func<-tibble::rownames_to_column(Aran_counts_func, var = "genome")

func_beta_aran <- aran_counts_func %>%
  pivot_longer(!genome,names_to="sample",values_to = "counts") %>%
  left_join(aran_transect, by = join_by(sample == sample)) %>%
  mutate(Elevation=factor(Elevation)) %>% 
  group_by(Elevation) %>% 
  group_split() %>% 
  map_dbl(., ~ .x %>%
            select(genome, sample, counts) %>%
            pivot_wider(names_from = sample, values_from = counts) %>%
            column_to_rownames(var = "genome") %>%
            tss() %>%
            as.data.frame() %>%
            hilldiss(data=., metric="S", q = 1, dist=dist)
  )

names(func_beta_aran)  <- unique(aran_transect$Elevation) %>% sort()
```

```{r aran_mean_1, eval=TRUE, results=TRUE}
# Merge all metrics
beta_div_aran <- bind_rows(richness_beta_aran,neutral_beta_aran,phylo_beta_aran,func_beta_aran) %>%
  t() %>% 
  as.data.frame() %>% 
  rownames_to_column(var="Elevation") %>% 
  as_tibble() %>% 
  rename("richness"=V1,"neutral"=V2,"phylogenetic"=V3,"functional"=V4)
```

#### Sentein

```{r sentein_mean, eval=FALSE}
sentein_transect<-tibble::rownames_to_column(sentein_transect, var = "sample")
sentein_counts<-tibble::rownames_to_column(sentein_counts, var = "genome")


richness_beta_sentein <- sentein_counts %>%
  pivot_longer(!genome,names_to="sample",values_to = "counts") %>%
  left_join(sentein_transect, by = join_by(sample == sample)) %>%
  mutate(Elevation=factor(Elevation)) %>% 
  group_by(Elevation) %>% 
  group_split() %>% 
  map_dbl(., ~ .x %>%
            select(genome, sample, counts) %>%
            pivot_wider(names_from = sample, values_from = counts) %>%
            column_to_rownames(var = "genome") %>%
            tss() %>%
            as.data.frame() %>%
            hilldiss(data=., metric="S", q = 0)
  )


names(richness_beta_sentein)  <- unique(sentein_transect$Elevation) %>% sort()

neutral_beta_sentein <- sentein_counts %>%
  pivot_longer(!genome,names_to="sample",values_to = "counts") %>%
  left_join(sentein_transect, by = join_by(sample == sample)) %>%
  mutate(Elevation=factor(Elevation)) %>% 
  group_by(Elevation) %>% 
  group_split() %>% 
  map_dbl(., ~ .x %>%
            select(genome, sample, counts) %>%
            pivot_wider(names_from = sample, values_from = counts) %>%
            column_to_rownames(var = "genome") %>%
            tss() %>%
            as.data.frame() %>%
            hilldiss(data=., metric="S", q = 1)
  )

names(neutral_beta_sentein)  <- unique(sentein_transect$Elevation) %>% sort()

phylo_beta_sentein <- sentein_counts %>%
  pivot_longer(!genome,names_to="sample",values_to = "counts") %>%
  left_join(sentein_transect, by = join_by(sample == sample)) %>%
  mutate(Elevation=factor(Elevation)) %>% 
  group_by(Elevation) %>% 
  group_split() %>% 
  map_dbl(., ~ .x %>%
            select(genome, sample, counts) %>%
            pivot_wider(names_from = sample, values_from = counts) %>%
            column_to_rownames(var = "genome") %>%
            tss() %>%
            as.data.frame() %>%
            hilldiss(data=., metric="S", q = 1, tree=genome_tree)
  )

names(phylo_beta_sentein)  <- unique(sentein_transect$Elevation) %>% sort()

sentein_counts_func<-tibble::rownames_to_column(sentein_counts_func, var = "genome")

func_beta_sentein <- sentein_counts_func %>%
  pivot_longer(!genome,names_to="sample",values_to = "counts") %>%
  left_join(sentein_transect, by = join_by(sample == sample)) %>%
  mutate(Elevation=factor(Elevation)) %>% 
  group_by(Elevation) %>% 
  group_split() %>% 
  map_dbl(., ~ .x %>%
            select(genome, sample, counts) %>%
            pivot_wider(names_from = sample, values_from = counts) %>%
            column_to_rownames(var = "genome") %>%
            tss() %>%
            as.data.frame() %>%
            hilldiss(data=., metric="S", q = 1, dist=dist)
  )

names(func_beta_sentein)  <- unique(sentein_transect$Elevation) %>% sort()
```


```{r sentein_mean_1, eval=TRUE, results=TRUE}
# Merge all metrics
beta_div_sentein <- bind_rows(richness_beta_sentein,neutral_beta_sentein,phylo_beta_sentein,func_beta_sentein) %>%
  t() %>% 
  as.data.frame() %>% 
  rownames_to_column(var="Elevation") %>% 
  as_tibble() %>% 
  rename("richness"=V1,"neutral"=V2,"phylogenetic"=V3,"functional"=V4)
```

#### Tourmalet

```{r tourmalet_mean, eval=FALSE}
tourmalet_transect<-tibble::rownames_to_column(tourmalet_transect, var = "sample")
tourmalet_counts<-tibble::rownames_to_column(tourmalet_counts, var = "genome")


richness_beta_tourmalet <- tourmalet_counts %>%
  pivot_longer(!genome,names_to="sample",values_to = "counts") %>%
  left_join(tourmalet_transect, by = join_by(sample == sample)) %>%
  mutate(Elevation=factor(Elevation)) %>% 
  group_by(Elevation) %>% 
  group_split() %>% 
  map_dbl(., ~ .x %>%
            select(genome, sample, counts) %>%
            pivot_wider(names_from = sample, values_from = counts) %>%
            column_to_rownames(var = "genome") %>%
            tss() %>%
            as.data.frame() %>%
            hilldiss(data=., metric="S", q = 0)
  )

names(richness_beta_tourmalet)  <- unique(tourmalet_transect$Elevation) %>% sort()

neutral_beta_tourmalet <- tourmalet_counts %>%
  pivot_longer(!genome,names_to="sample",values_to = "counts") %>%
  left_join(tourmalet_transect, by = join_by(sample == sample)) %>%
  mutate(Elevation=factor(Elevation)) %>% 
  group_by(Elevation) %>% 
  group_split() %>% 
  map_dbl(., ~ .x %>%
            select(genome, sample, counts) %>%
            pivot_wider(names_from = sample, values_from = counts) %>%
            column_to_rownames(var = "genome") %>%
            tss() %>%
            as.data.frame() %>%
            hilldiss(data=., metric="S", q = 1)
  )

names(neutral_beta_tourmalet)  <- unique(tourmalet_transect$Elevation) %>% sort()

phylo_beta_tourmalet <- tourmalet_counts %>%
  pivot_longer(!genome,names_to="sample",values_to = "counts") %>%
  left_join(tourmalet_transect, by = join_by(sample == sample)) %>%
  mutate(Elevation=factor(Elevation)) %>% 
  group_by(Elevation) %>% 
  group_split() %>% 
  map_dbl(., ~ .x %>%
            select(genome, sample, counts) %>%
            pivot_wider(names_from = sample, values_from = counts) %>%
            column_to_rownames(var = "genome") %>%
            tss() %>%
            as.data.frame() %>%
            hilldiss(data=., metric="S", q = 1, tree=genome_tree)
  )

names(phylo_beta_tourmalet)  <- unique(tourmalet_transect$Elevation) %>% sort()

tourmalet_counts_func<-tibble::rownames_to_column(tourmalet_counts_func, var = "genome")

func_beta_tourmalet <- tourmalet_counts_func %>%
  pivot_longer(!genome,names_to="sample",values_to = "counts") %>%
  left_join(tourmalet_transect, by = join_by(sample == sample)) %>%
  mutate(Elevation=factor(Elevation)) %>% 
  group_by(Elevation) %>% 
  group_split() %>% 
  map_dbl(., ~ .x %>%
            select(genome, sample, counts) %>%
            pivot_wider(names_from = sample, values_from = counts) %>%
            column_to_rownames(var = "genome") %>%
            tss() %>%
            as.data.frame() %>%
            hilldiss(data=., metric="S", q = 1, dist=dist)
  )

names(func_beta_tourmalet)  <- unique(tourmalet_transect$Elevation) %>% sort()
```


```{r tourmalet_mean_1, eval=TRUE, results=TRUE}
# Merge all metrics
beta_div_tourmalet <- bind_rows(richness_beta_tourmalet,neutral_beta_tourmalet,phylo_beta_tourmalet,func_beta_tourmalet) %>%
  t() %>% 
  as.data.frame() %>% 
  rownames_to_column(var="Elevation") %>% 
  as_tibble() %>% 
  rename("richness"=V1,"neutral"=V2,"phylogenetic"=V3,"functional"=V4)
```

