# Diversity analyses

```{r load_data_community_1}
load("data/data.Rdata")
load("data/alpha_div_21032025.Rdata")
load("data/beta_div_21032025.Rdata")
load("data/beta_div_pop_21032025.Rdata")
```

## Alpha diversity

```{r alpha_div, comment="", message=FALSE, warning=FALSE, eval=FALSE}
# Calculate Hill numbers
richness <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 0) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(richness = 1) %>%
  rownames_to_column(var = "sample")

neutral <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(neutral = 1) %>%
  rownames_to_column(var = "sample")

phylogenetic <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1, tree = genome_tree) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(phylogenetic = 1) %>%
  rownames_to_column(var = "sample")

# Aggregate basal GIFT into elements
#Get list of present MAGs
present_MAGs <- genome_counts_filt %>%
    column_to_rownames(var = "genome") %>%
		filter(rowSums(.[, -1]) != 0) %>%
		rownames()

#Align KEGG annotations with present MAGs and remove all-zero and all-one traits
present_MAGs <- present_MAGs[present_MAGs %in% rownames(genome_gifts)]
genome_gifts_filt <- genome_gifts[present_MAGs,] %>%
			select_if(~!all(. == 0)) %>%  #remove all-zero modules
			select_if(~!all(. == 1)) #remove all-one modules

#Filter count table to only contain present MAGs after KEGG filtering
genome_counts_filt_filt <- genome_counts_filt %>%
  column_to_rownames(var = "genome")

genome_counts_filt_filt <- genome_counts_filt_filt[present_MAGs,]

# Merge all metrics
alpha_div <- richness %>%
  full_join(neutral, by = join_by(sample == sample)) %>%
  full_join(phylogenetic, by = join_by(sample == sample))

save(genome_gifts_filt,
      richness,
     neutral,
     phylogenetic,
     alpha_div,
     file = "data/alpha_div_21032025.Rdata")
```

```{r alpha_div_summary, comment="", message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
  pivot_longer(-sample, names_to = "alpha", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == EHI_number)) %>%
    group_by(alpha)%>%
    summarise(
              Aisa_mean=mean(value[Transect=="Aisa"], na.rm=T),
              Aisa_sd=sd(value[Transect=="Aisa"], na.rm=T),
              Aran_mean=mean(value[Transect=="Aran"], na.rm=T),
              Aran_sd=sd(value[Transect=="Aran"], na.rm=T),
              Sentein_mean=mean(value[Transect=="Sentein"], na.rm=T),
              Sentein_sd=sd(value[Transect=="Sentein"], na.rm=T),
              Tourmalet_mean=mean(value[Transect=="Tourmalet"], na.rm=T),
              Tourmalet_sd=sd(value[Transect=="Tourmalet"], na.rm=T))%>%
    mutate(
           Aisa=str_c(round(Aisa_mean,2),"±",round(Aisa_sd,2)),
           Aran=str_c(round(Aran_mean,2),"±",round(Aran_sd,2)),
           Sentein=str_c(round(Sentein_mean,2),"±",round(Sentein_sd,2)),
           Tourmalet=str_c(round(Tourmalet_mean,2),"±",round(Tourmalet_sd,2))) %>% 
    arrange(-Aisa_mean) %>% 
    dplyr::select(alpha,Aisa,Aran,Sentein,Tourmalet) %>% 
    tt()
```

```{r alpha_div_plot, comment="", message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == EHI_number)) %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic"))) %>%
      ggplot(aes(y = value, x = Transect, group=Transect, color=Transect, fill=Transect)) +
      geom_boxplot(outlier.shape = NA) +
      geom_jitter(alpha=0.5) +
      scale_color_manual(name="Transect",
          breaks=c("Aisa","Aran","Sentein","Tourmalet"),
          labels=c("Aisa","Aran","Sentein","Tourmalet"),
          values=c("#e5bd5b", "#6b7398","#e2815a", "#876b96")) +
      scale_fill_manual(name="Transect",
          breaks=c("Aisa","Aran","Sentein","Tourmalet"),
          labels=c("Aisa","Aran","Sentein","Tourmalet"),
          values=c("#e5bd5b50", "#6b739850","#e2815a50", "#876b9650")) +
      facet_wrap(. ~ metric, scales = "free", ncol=4) +
      coord_cartesian(xlim = c(1, NA)) +
      stat_compare_means(size=2) +
      theme_classic() +
      theme(
        strip.background = element_blank(),
        panel.grid.minor.x = element_line(size = .1, color = "grey"),
        axis.text.x = element_text(angle = 45, hjust = 1)
      ) +
  ylab("Alpha diversity")      
      
```

### Regression plots per transect

```{r alpha_div_richness_plot, message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == EHI_number)) %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic"))) %>%
  filter(metric!="functional")%>%
      ggplot(aes(y = value, x = Elevation, group=Transect, colour=Transect)) +
      geom_point(alpha=0.4) +
        geom_smooth(method = "loess", se=FALSE) +
      scale_colour_manual(name="Transect",
          breaks=c("Aisa","Aran","Sentein","Tourmalet"),
          labels=c("Aisa","Aran","Sentein","Tourmalet"),
          values=c("#e5bd5b", "#6b7398","#e2815a", "#876b96")) +
      facet_wrap(. ~ metric, scales = "free", ncol=4) +
      coord_cartesian(xlim = c(1, NA)) +
      theme_classic() +
      theme(
        strip.background = element_blank(),
        panel.grid.minor.x = element_line(size = .1, color = "grey"),
        axis.text.x = element_text(angle = 45, hjust = 1)
      ) + 
  ylab("Alpha diversity")
```


### Mixed models per metric

```{r mixed_models_data, echo=TRUE, message=FALSE, warning=FALSE}
alpha_div_meta<-alpha_div %>%
  left_join(sample_metadata, by = join_by(sample == EHI_number))
```

#### Richness diversity

```{r mixed_models_richness, echo=TRUE, message=FALSE, warning=FALSE}
Modelq0GLMMNB<- lmerTest::lmer(richness ~ log(sequencing_depth) + Elevation+ I(Elevation^2)+Transect + (1 | Sampling_point), data = alpha_div_meta, REML = FALSE)
summary(Modelq0GLMMNB)
emmeans::emmeans(Modelq0GLMMNB, pairwise ~ Transect)
```

#### Neutral diversity

```{r mixed_models_neutral, echo=TRUE, message=FALSE, warning=FALSE}
Modelq1n <- nlme::lme(neutral ~ log(sequencing_depth) + Elevation+I(Elevation^2)+Transect, 
               random = ~ 1 | Sampling_point,
               data = alpha_div_meta)
summary(Modelq1n)
emmeans::emmeans(Modelq1n, pairwise ~ Transect)
```

#### Phylogenetic diversity

```{r mixed_models_phylo, echo=TRUE, message=FALSE, warning=FALSE}
Modelq1p <- nlme::lme(phylogenetic ~ log(sequencing_depth) + Elevation+I(Elevation^2)+Transect, 
               random = ~ 1 | Sampling_point,
               data = alpha_div_meta)
summary(Modelq1p)
emmeans::emmeans(Modelq1p, pairwise ~ Transect)
```

## Beta diversity per individual

```{r beta_div, comment="", message=FALSE, warning=FALSE, eval=FALSE}
beta_q0n <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  hillpair(., q = 0)

beta_q1n <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  hillpair(., q = 1)

beta_q1p <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  hillpair(., q = 1, tree = genome_tree)

save(beta_q0n,
     beta_q1n,
     beta_q1p,
     file = "data/beta_div_21032025.Rdata")
```

#### Richness

```{r beta_div_richness_permutest, echo=TRUE, results = TRUE}
betadisper(beta_q0n$S, sample_metadata$Transect) %>% permutest(., pairwise=TRUE) 
```

```{r beta_div_richness_adonis, echo=TRUE, results = TRUE}
adonis2(beta_q0n$S ~ Elevation+I(Elevation^2)+Transect, 
        data = sample_metadata %>% arrange(match(EHI_number,labels(beta_q0n$S))), 
        permutations = 999,
        strata = sample_metadata %>% arrange(match(EHI_number,labels(beta_q0n$S))) %>% pull(Sampling_point),
        by="terms") %>%
        broom::tidy() %>%
        tt()
```

#### Neutral

```{r beta_div_neutral_permutest, echo=TRUE, results = TRUE}
betadisper(beta_q1n$S, sample_metadata$Transect) %>% permutest(., pairwise=TRUE) 
```

```{r beta_div_neutral_adonis, echo=TRUE, results = TRUE}
adonis2(beta_q1n$S ~ Elevation+I(Elevation^2)+Transect+Sampling_point, 
        data = sample_metadata %>% arrange(match(EHI_number,labels(beta_q1n$S))), 
        permutations = 999,
        strata = sample_metadata %>% arrange(match(EHI_number,labels(beta_q1n$S))) %>% pull(Sampling_point),
        by="terms") %>%
        broom::tidy() %>%
        tt()
```

#### Phylogenetic

```{r beta_div_phylo_permutest, echo=TRUE, results = TRUE}
betadisper(beta_q1p$S, sample_metadata$Transect) %>% permutest(., pairwise=TRUE) 
```

```{r beta_div_phylo_adonis, echo=TRUE, results = TRUE}
adonis2(beta_q1p$S ~ Elevation+I(Elevation^2)+Transect, 
        data = sample_metadata %>% arrange(match(EHI_number,labels(beta_q1p$S))), 
        permutations = 999,
        strata = sample_metadata %>% arrange(match(EHI_number,labels(beta_q1p$S))) %>% pull(Sampling_point),
        by="terms") %>%
        broom::tidy() %>%
        tt()
```

## Beta diversity per population

```{r beta_div_transects, comment="", message=FALSE, warning=FALSE, eval=FALSE}
richness_beta <- genome_counts_filt %>%
  pivot_longer(!genome,names_to="sample",values_to = "counts") %>%
  left_join(sample_metadata, by = join_by(sample == EHI_number)) %>%
  mutate(Sampling_point=factor(Sampling_point)) %>% 
  group_by(Sampling_point) %>% 
  group_split() %>% 
  map_dbl(., ~ .x %>%
    select(genome, sample, counts) %>%
    pivot_wider(names_from = sample, values_from = counts) %>%
    column_to_rownames(var = "genome") %>%
    tss() %>%
    as.data.frame() %>%
    hilldiss(data=., metric="S", q = 0)
  )
names(richness_beta)  <- unique(sample_metadata$Sampling_point) %>% sort()


neutral_beta <- genome_counts_filt %>%
  pivot_longer(!genome,names_to="sample",values_to = "counts") %>%
  left_join(sample_metadata, by = join_by(sample == EHI_number)) %>%
  mutate(Sampling_point=factor(Sampling_point)) %>% 
  group_by(Sampling_point) %>% 
  group_split() %>% 
  map_dbl(., ~ .x %>%
    select(genome, sample, counts) %>%
    pivot_wider(names_from = sample, values_from = counts) %>%
    column_to_rownames(var = "genome") %>%
    tss() %>%
    as.data.frame() %>%
    hilldiss(data=., metric="S", q = 1)
  )
names(neutral_beta)  <- unique(sample_metadata$Sampling_point) %>% sort()

phylogenetic_beta <- genome_counts_filt %>%
  pivot_longer(!genome,names_to="sample",values_to = "counts") %>%
  left_join(sample_metadata, by = join_by(sample == EHI_number)) %>%
  mutate(Sampling_point=factor(Sampling_point)) %>% 
  group_by(Sampling_point) %>% 
  group_split() %>% 
  map_dbl(., ~ .x %>%
    select(genome, sample, counts) %>%
    pivot_wider(names_from = sample, values_from = counts) %>%
    column_to_rownames(var = "genome") %>%
    tss() %>%
    as.data.frame() %>%
    hilldiss(data=., metric="S", q = 1, tree = genome_tree)
  )
names(phylogenetic_beta)  <- unique(sample_metadata$Sampling_point) %>% sort()

# Merge all metrics
beta_div <- bind_rows(richness_beta,neutral_beta,phylogenetic_beta) %>%
  t() %>% 
  as.data.frame() %>% 
  rownames_to_column(var="Sampling_point") %>% 
  as_tibble() %>% 
  rename("richness"=V1,"neutral"=V2,"phylogenetic"=V3)

save(richness_beta,
     neutral_beta,
     phylogenetic_beta,
     beta_div,
     file = "data/beta_div_pop_21032025.Rdata")
```


```{r beta_div_boxplot, comment="", message=FALSE, warning=FALSE, fig.height=5, fig.width=10, fig.fullwidth=TRUE}
beta_div %>%
  pivot_longer(-Sampling_point, names_to = "metric", values_to = "value") %>%
  left_join(sample_metadata %>% select(Transect,Sampling_point, Elevation) %>% unique, by = "Sampling_point") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic"))) %>%
      ggplot(aes(y = value, x = Transect,group=Transect, fill=Transect)) +
      geom_boxplot(outlier.shape = NA) +
      geom_jitter(aes(color = Elevation), alpha = 0.5, size = 2) +
      scale_colour_gradient(low="#a7f0ba", high="#044319") +
      scale_fill_manual(name="Transect",
          breaks=c("Aisa","Aran","Sentein","Tourmalet"),
          labels=c("Aisa","Aran","Sentein","Tourmalet"),
          values=c("#e5bd5b50", "#6b739850","#e2815a50", "#876b9650")) +
      facet_wrap(. ~ metric, scales = "free", ncol=4) +
      coord_cartesian(xlim = c(1, NA)) +
      stat_compare_means(size=2) +
      theme_classic() +
      theme(
        strip.background = element_blank(),
        panel.grid.minor.x = element_line(size = .1, color = "grey"),
        axis.text.x = element_text(angle = 45, hjust = 1)
      ) + 
  ylab("Beta diversity")

```

```{r beta_div_boxplot_1, comment="", results=FALSE, message=FALSE, warning=FALSE, fig.height=5, fig.width=10, fig.fullwidth=TRUE}
beta_div_met<-beta_div %>%
  pivot_longer(-Sampling_point, names_to = "metric", values_to = "value") %>%
  left_join(sample_metadata %>% select(Transect,Sampling_point, Elevation)) %>%
  mutate(Zones = case_when(
    Transect == "Aisa" & Elevation == 1450 ~ "Z0",
    Transect == "Aisa" & Elevation == 1650 ~ "Z1",
    Transect == "Aisa" & Elevation == 1850 ~ "Z2",
    Transect == "Aran" & Elevation == 1000 ~ "Z0",
    Transect == "Aran" & Elevation == 1500 ~ "Z1",
    Transect == "Aran" & Elevation == 1650 ~ "Z2",
    Transect == "Aran" & Elevation == 1850 ~ "Z3",
    Transect == "Sentein" & Elevation == 941 ~ "Z0",
    Transect == "Sentein" & Elevation == 1260 ~ "Z1",
    Transect == "Sentein" & Elevation == 1628 ~ "Z2",
    Transect == "Sentein" & Elevation == 1873 ~ "Z3",
    Transect == "Tourmalet" & Elevation == 953 ~ "Z0",
    Transect == "Tourmalet" & Elevation == 1255 ~ "Z1",
    Transect == "Tourmalet" & Elevation == 1797 ~ "Z2",
    Transect == "Tourmalet" & Elevation == 2306 ~ "Z3",
    TRUE ~ NA_character_)) %>%
  filter(Zones!="NA") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic","functional")))


beta_div_met$Zones<-as.factor(beta_div_met$Zones)
beta_div_met$Tras<-as.factor(beta_div_met$Zones)
```

```{r beta_div_boxplot_2, comment="", results=FALSE, message=FALSE, warning=FALSE, fig.height=15, fig.width=14}
bxp1<-beta_div_met%>%
      ggplot(aes(y = value, x = Zones, group=Zones, colour=Transect, fill=Transect)) +
      geom_boxplot() +
      geom_jitter() +
      #scale_colour_gradient(low="#a7f0ba", high="#044319") +
      scale_colour_manual(name="Transect",
          breaks=c("Aisa","Aran","Sentein","Tourmalet"),
          labels=c("Aisa","Aran","Sentein","Tourmalet"),
          values=c("#e5bd5b", "#6b7398","#e2815a", "#876b96")) +
      scale_fill_manual(name="Transect",
          breaks=c("Aisa","Aran","Sentein","Tourmalet"),
          labels=c("Aisa","Aran","Sentein","Tourmalet"),
          values=c("#e5bd5b50", "#6b739850","#e2815a50", "#876b9650")) +
      facet_wrap(. ~ metric+Transect, scales = "free", ncol=4) +
      coord_cartesian(xlim = c(1, NA)) +
      stat_compare_means(size=2) +
      theme_classic() +
      theme(
        strip.background = element_blank(),
        panel.grid.minor.x = element_line(size = .1, color = "grey"),
        axis.text.x = element_text(angle = 45, hjust = 1)
      ) + 
  ylab("Beta diversity")

# Add significance bars with p-values
bxp2<-bxp1 + 
  geom_signif(comparisons = list(c("Z0", "Z1"), c("Z1", "Z2"), c("Z2", "Z3"), c("Z0", "Z2"), c("Z1", "Z3")),
              map_signif_level = TRUE)

bxp2+geom_signif(comparisons = list(c("Z0", "Z1"), c("Z1", "Z2")),
              map_signif_level = TRUE)
```

### Transit zones

```{r beta_div_pop_data, comment="", results=FALSE, message=FALSE, warning=FALSE}
beta_div_pop<-beta_div %>%
  pivot_longer(-Sampling_point, names_to = "metric", values_to = "value") %>%
  left_join(sample_metadata %>% select(Transect,Sampling_point, Elevation) %>% unique, by = "Sampling_point")
beta_div_pop <- beta_div_pop %>%
  mutate(Zones = case_when(
    Transect == "Aisa" & Elevation == 1450 ~ "Z0",
    Transect == "Aisa" & Elevation == 1650 ~ "Z1",
    Transect == "Aisa" & Elevation == 1850 ~ "Z2",
    Transect == "Aran" & Elevation == 1000 ~ "Z0",
    Transect == "Aran" & Elevation == 1500 ~ "Z1",
    Transect == "Aran" & Elevation == 1650 ~ "Z2",
    Transect == "Aran" & Elevation == 1850 ~ "Z3",
    Transect == "Sentein" & Elevation == 941 ~ "Z0",
    Transect == "Sentein" & Elevation == 1260 ~ "Z1",
    Transect == "Sentein" & Elevation == 1628 ~ "Z2",
    Transect == "Sentein" & Elevation == 1873 ~ "Z3",
    Transect == "Tourmalet" & Elevation == 953 ~ "Z0",
    Transect == "Tourmalet" & Elevation == 1255 ~ "Z1",
    Transect == "Tourmalet" & Elevation == 1797 ~ "Z2",
    Transect == "Tourmalet" & Elevation == 2306 ~ "Z3",
    TRUE ~ NA_character_)) %>%
  filter(Zones!="NA")
```

```{r beta_div_boxplot_3, comment="", results=FALSE, message=FALSE, warning=FALSE, fig.height=5, fig.width=10, fig.fullwidth=TRUE}
beta_div_pop %>%
  #pivot_longer(-Sampling_point, names_to = "metric", values_to = "value") %>%
  #left_join(sample_metadata %>% select(Transect,Sampling_point, Elevation) %>% unique, by = "Sampling_point") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic"))) %>%
      ggplot(aes(y = value, x = Elevation ,group=Elevation, colour=Transect)) +
      #geom_boxplot(outlier.shape = NA) +
      geom_point(aes(color = Transect), alpha = 0.5, size = 2) + 
      geom_line(aes(group = interaction(Transect, metric)), position = position_dodge(width = 0.5))+
      #scale_color_viridis_c(name = "Elevation", option = "viridis") +
      scale_color_manual(name="Transect",
          breaks=c("Aisa","Aran","Sentein","Tourmalet"),
          labels=c("Aisa","Aran","Sentein","Tourmalet"),
          values=c("#e5bd5b50", "#6b739850","#e2815a50", "#876b9650")) +
      facet_wrap(. ~ metric, scales = "free", ncol=4) +
      coord_cartesian(xlim = c(1, NA)) +
      #stat_compare_means(size=2) +
      theme_classic() +
      theme(
        strip.background = element_blank(),
        panel.grid.minor.x = element_line(size = .1, color = "grey"),
        axis.text.x = element_text(angle = 45, hjust = 1)
      ) + 
  ylab("Beta diversity")

```

#### Aisa

```{r beta_div_pop_aisa_data, warning=FALSE, comments="", message=FALSE}
samples_to_keep_aisa <- sample_metadata %>%
  filter(Transect == "Aisa") %>% 
  filter(Elevation!=1250)%>% 
  select(EHI_number) %>% 
  pull()
subset_meta_aisa <- sample_metadata %>%
  filter(Transect == "Aisa")%>% 
  filter(Elevation!=1250)%>%
  mutate(Zones = case_when(
    Transect == "Aisa" & Elevation == 1450 ~ "Z0",
    Transect == "Aisa" & Elevation == 1650 ~ "Z1",
    Transect == "Aisa" & Elevation == 1850 ~ "Z2"))
```

***Richness***
```{r beta_div_pop_aisa, warning=FALSE, comments="", message=FALSE}
richness_aisa<-as.matrix(beta_q0n$S)
richness_aisa <- as.dist(richness_aisa[rownames(richness_aisa) %in% samples_to_keep_aisa,
               colnames(richness_aisa) %in% samples_to_keep_aisa])
betadisper(richness_aisa, subset_meta_aisa$Zones) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_aisa, warning=FALSE, comments="", message=FALSE}
adonis2(richness_aisa ~ Zones,
        data = subset_meta_aisa %>% arrange(match(EHI_number,labels(richness_aisa))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_pop_aisa_1, warning=FALSE, comments="", message=FALSE}
pairwise <- pairwise.adonis(richness_aisa, subset_meta_aisa$Zones, perm=999)
pairwise%>%
  tt()
```

***Neutral***
```{r beta_div_pop_neut_aisa, warning=FALSE, comments="", message=FALSE}
neutral_aisa<-as.matrix(beta_q1n$S)
neutral_aisa <- as.dist(neutral_aisa[rownames(neutral_aisa) %in% samples_to_keep_aisa,
               colnames(neutral_aisa) %in% samples_to_keep_aisa])
betadisper(neutral_aisa, subset_meta_aisa$Zones) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_neut_aisa, warning=FALSE, comments="", message=FALSE}
adonis2(neutral_aisa ~ Zones,
        data = subset_meta_aisa %>% arrange(match(EHI_number,labels(neutral_aisa))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_pop_aisa_neut_1, warning=FALSE, comments="", message=FALSE}
pairwise <- pairwise.adonis(neutral_aisa, subset_meta_aisa$Zones, perm=999)
pairwise%>%
  tt()
```

***Phylogenetic***
```{r beta_div_pop_phylo_aisa, warning=FALSE, comments="", message=FALSE}
phylo_aisa<-as.matrix(beta_q1p$S)
phylo_aisa <- as.dist(phylo_aisa[rownames(phylo_aisa) %in% samples_to_keep_aisa,
               colnames(phylo_aisa) %in% samples_to_keep_aisa])
betadisper(phylo_aisa, subset_meta_aisa$Zones) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_phylo_aisa, warning=FALSE, comments="", message=FALSE}
adonis2(phylo_aisa ~ Zones,
        data = subset_meta_aisa %>% arrange(match(EHI_number,labels(phylo_aisa))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_pop_aisa_phylo_1, warning=FALSE, comments="", message=FALSE}
pairwise <- pairwise.adonis(phylo_aisa, subset_meta_aisa$Zones, perm=999)
pairwise%>%
  tt()
```

***Functional***
```{r beta_div_pop_func_aisa, warning=FALSE, comments="", message=FALSE}
func_aisa<-as.matrix(beta_q0n$S)
func_aisa <- as.dist(func_aisa[rownames(func_aisa) %in% samples_to_keep_aisa,
               colnames(func_aisa) %in% samples_to_keep_aisa])
betadisper(func_aisa, subset_meta_aisa$Zones) %>% permutest(., pairwise = TRUE)
```

```{r beta_func_permanova_aisa, warning=FALSE, comments="", message=FALSE}
adonis2(func_aisa ~ Zones,
        data = subset_meta_aisa %>% arrange(match(EHI_number,labels(func_aisa))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_pop_aisa_func, warning=FALSE, comments="", message=FALSE}
pairwise <- pairwise.adonis(func_aisa, subset_meta_aisa$Zones, perm=999)
pairwise%>%
  tt()
```

#### Aran

```{r beta_div_pop_Aran_data, warning=FALSE, comments="", message=FALSE}
samples_to_keep_Aran <- sample_metadata %>%
  filter(Transect == "Aran") %>% 
  filter(Elevation!=1080) %>%
  filter(Elevation!=1340) %>% 
  select(EHI_number) %>% 
  pull()
subset_meta_Aran <- sample_metadata %>%
  filter(Transect == "Aran")%>% 
filter(Elevation!=1080) %>%
filter(Elevation!=1340) %>% 
  mutate(Zones = case_when(
    Transect == "Aran" & Elevation == 1000 ~ "Z0",
    Transect == "Aran" & Elevation == 1500 ~ "Z1",
    Transect == "Aran" & Elevation == 1650 ~ "Z2",
    Transect == "Aran" & Elevation == 1850 ~ "Z3"))
```

***Richness***
```{r beta_div_pop_Aran, warning=FALSE, comments="", message=FALSE}
richness_Aran<-as.matrix(beta_q0n$S)
richness_Aran <- as.dist(richness_Aran[rownames(richness_Aran) %in% samples_to_keep_Aran,
               colnames(richness_Aran) %in% samples_to_keep_Aran])
betadisper(richness_Aran, subset_meta_Aran$Zones) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_Aran, warning=FALSE, comments="", message=FALSE}
adonis2(richness_Aran ~ Zones,
        data = subset_meta_Aran %>% arrange(match(EHI_number,labels(richness_Aran))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_pop_Aran_1, warning=FALSE, comments="", message=FALSE}
pairwise <- pairwise.adonis(richness_Aran, subset_meta_Aran$Zones, perm=999)
pairwise%>%
  tt()
```

***Neutral***
```{r beta_div_pop_neut_Aran, warning=FALSE, comments="", message=FALSE}
neutral_Aran<-as.matrix(beta_q1n$S)
neutral_Aran <- as.dist(neutral_Aran[rownames(neutral_Aran) %in% samples_to_keep_Aran,
               colnames(neutral_Aran) %in% samples_to_keep_Aran])
betadisper(neutral_Aran, subset_meta_Aran$Zones) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_neut_Aran, warning=FALSE, comments="", message=FALSE}
adonis2(neutral_Aran ~ Zones,
        data = subset_meta_Aran %>% arrange(match(EHI_number,labels(neutral_Aran))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_pop_Aran_neut_1, warning=FALSE, comments="", message=FALSE}
pairwise <- pairwise.adonis(neutral_Aran, subset_meta_Aran$Zones, perm=999)
pairwise%>%
  tt()
```

***Phylogenetic***
```{r beta_div_pop_phylo_Aran, warning=FALSE, comments="", message=FALSE}
phylo_Aran<-as.matrix(beta_q1p$S)
phylo_Aran <- as.dist(phylo_Aran[rownames(phylo_Aran) %in% samples_to_keep_Aran,
               colnames(phylo_Aran) %in% samples_to_keep_Aran])
betadisper(phylo_Aran, subset_meta_Aran$Zones) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_phylo_Aran, warning=FALSE, comments="", message=FALSE}
adonis2(phylo_Aran ~ Zones,
        data = subset_meta_Aran %>% arrange(match(EHI_number,labels(phylo_Aran))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_pop_Aran_phylo_1, warning=FALSE, comments="", message=FALSE}
pairwise <- pairwise.adonis(phylo_Aran, subset_meta_Aran$Zones, perm=999)
pairwise%>%
  tt()
```

***Functional***
```{r beta_div_pop_func_Aran, warning=FALSE, comments="", message=FALSE}
func_Aran<-as.matrix(beta_q0n$S)
func_Aran <- as.dist(func_Aran[rownames(func_Aran) %in% samples_to_keep_Aran,
               colnames(func_Aran) %in% samples_to_keep_Aran])
betadisper(func_Aran, subset_meta_Aran$Zones) %>% permutest(., pairwise = TRUE)
```

```{r beta_func_permanova_Aran, warning=FALSE, comments="", message=FALSE}
adonis2(func_Aran ~ Zones,
        data = subset_meta_Aran %>% arrange(match(EHI_number,labels(func_Aran))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_pop_Aran_1_func, warning=FALSE, comments="", message=FALSE}
pairwise <- pairwise.adonis(func_Aran, subset_meta_Aran$Zones, perm=999)
pairwise%>%
  tt()
```

#### Sentein

```{r beta_div_pop_Sentein_data, warning=FALSE, comments="", message=FALSE}
samples_to_keep_Sentein <- sample_metadata %>%
  filter(Transect == "Sentein") %>% 
  select(EHI_number) %>% 
  pull()
subset_meta_Sentein <- sample_metadata %>%
  filter(Transect == "Sentein")%>% 
  mutate(Zones = case_when(
    Transect == "Sentein" & Elevation == 941 ~ "Z0",
    Transect == "Sentein" & Elevation == 1260 ~ "Z1",
    Transect == "Sentein" & Elevation == 1628 ~ "Z2",
    Transect == "Sentein" & Elevation == 1873 ~ "Z3"))
```

***Richness***
```{r beta_div_pop_Sentein, warning=FALSE, comments="", message=FALSE}
richness_Sentein<-as.matrix(beta_q0n$S)
richness_Sentein <- as.dist(richness_Sentein[rownames(richness_Sentein) %in% samples_to_keep_Sentein,
               colnames(richness_Sentein) %in% samples_to_keep_Sentein])
betadisper(richness_Sentein, subset_meta_Sentein$Zones) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_Sentein, warning=FALSE, comments="", message=FALSE}
adonis2(richness_Sentein ~ Zones,
        data = subset_meta_Sentein %>% arrange(match(EHI_number,labels(richness_Sentein))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_pop_Sentein_1, warning=FALSE, comments="", message=FALSE}
pairwise <- pairwise.adonis(richness_Sentein, subset_meta_Sentein$Zones, perm=999)
pairwise%>%
  tt()
```

***Neutral***
```{r beta_div_pop_neut_Sentein, warning=FALSE, comments="", message=FALSE}
neutral_Sentein<-as.matrix(beta_q1n$S)
neutral_Sentein <- as.dist(neutral_Sentein[rownames(neutral_Sentein) %in% samples_to_keep_Sentein,
               colnames(neutral_Sentein) %in% samples_to_keep_Sentein])
betadisper(neutral_Sentein, subset_meta_Sentein$Zones) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_neut_Sentein, warning=FALSE, comments="", message=FALSE}
adonis2(neutral_Sentein ~ Zones,
        data = subset_meta_Sentein %>% arrange(match(EHI_number,labels(neutral_Sentein))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_pop_Sentein_neut_1, warning=FALSE, comments="", message=FALSE}
pairwise <- pairwise.adonis(neutral_Sentein, subset_meta_Sentein$Zones, perm=999)
pairwise%>%
  tt()
```

***Phylogenetic***
```{r beta_div_pop_phylo_Sentein, warning=FALSE, comments="", message=FALSE}
phylo_Sentein<-as.matrix(beta_q1p$S)
phylo_Sentein <- as.dist(phylo_Sentein[rownames(phylo_Sentein) %in% samples_to_keep_Sentein,
               colnames(phylo_Sentein) %in% samples_to_keep_Sentein])
betadisper(phylo_Sentein, subset_meta_Sentein$Zones) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_phylo_Sentein, warning=FALSE, comments="", message=FALSE}
adonis2(phylo_Sentein ~ Zones,
        data = subset_meta_Sentein %>% arrange(match(EHI_number,labels(phylo_Sentein))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_pop_Sentein_phylo_1, warning=FALSE, comments="", message=FALSE}
pairwise <- pairwise.adonis(phylo_Sentein, subset_meta_Sentein$Zones, perm=999)
pairwise%>%
  tt()
```

***Functional***
```{r beta_div_pop_func_Sentein, warning=FALSE, comments="", message=FALSE}
func_Sentein<-as.matrix(beta_q0n$S)
func_Sentein <- as.dist(func_Sentein[rownames(func_Sentein) %in% samples_to_keep_Sentein,
               colnames(func_Sentein) %in% samples_to_keep_Sentein])
betadisper(func_Sentein, subset_meta_Sentein$Zones) %>% permutest(., pairwise = TRUE)
```

```{r beta_func_permanova_Sentein, warning=FALSE, comments="", message=FALSE}
adonis2(func_Sentein ~ Zones,
        data = subset_meta_Sentein %>% arrange(match(EHI_number,labels(func_Sentein))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_pop_Sentein_1_func, warning=FALSE, comments="", message=FALSE}
pairwise <- pairwise.adonis(func_Sentein, subset_meta_Sentein$Zones, perm=999)
pairwise%>%
  tt()
```

#### Tourmalet

```{r beta_div_pop_Tourmalet_data, warning=FALSE, comments="", message=FALSE}
samples_to_keep_Tourmalet <- sample_metadata %>%
  filter(Transect == "Tourmalet") %>% 
  filter(Elevation!=1561)%>% 
  filter(Elevation!=2065)%>% 
  filter(Elevation!=2134)%>% 
  select(EHI_number) %>% 
  pull()
subset_meta_Tourmalet <- sample_metadata %>%
  filter(Transect == "Tourmalet")%>% 
   filter(Elevation!=1561)%>% 
  filter(Elevation!=2065)%>% 
  filter(Elevation!=2134)%>%
  mutate(Zones = case_when(
    Transect == "Tourmalet" & Elevation == 953 ~ "Z0",
    Transect == "Tourmalet" & Elevation == 1255 ~ "Z1",
    Transect == "Tourmalet" & Elevation == 1797 ~ "Z2",
    Transect == "Tourmalet" & Elevation == 2306 ~ "Z3"))
```

***Richness***
```{r beta_div_pop_Tourmalet, warning=FALSE, comments="", message=FALSE}
richness_Tourmalet<-as.matrix(beta_q0n$S)
richness_Tourmalet <- as.dist(richness_Tourmalet[rownames(richness_Tourmalet) %in% samples_to_keep_Tourmalet,
               colnames(richness_Tourmalet) %in% samples_to_keep_Tourmalet])
betadisper(richness_Tourmalet, subset_meta_Tourmalet$Zones) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_Tourmalet, warning=FALSE, comments="", message=FALSE}
adonis2(richness_Tourmalet ~ Zones,
        data = subset_meta_Tourmalet %>% arrange(match(EHI_number,labels(richness_Tourmalet))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_pop_Tourmalet_1, warning=FALSE, comments="", message=FALSE}
pairwise <- pairwise.adonis(richness_Tourmalet, subset_meta_Tourmalet$Zones, perm=999)
pairwise%>%
  tt()
```

***Neutral***
```{r beta_div_pop_neut_Tourmalet, warning=FALSE, comments="", message=FALSE}
neutral_Tourmalet<-as.matrix(beta_q1n$S)
neutral_Tourmalet <- as.dist(neutral_Tourmalet[rownames(neutral_Tourmalet) %in% samples_to_keep_Tourmalet,
               colnames(neutral_Tourmalet) %in% samples_to_keep_Tourmalet])
betadisper(neutral_Tourmalet, subset_meta_Tourmalet$Zones) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_neut_Tourmalet, warning=FALSE, comments="", message=FALSE}
adonis2(neutral_Tourmalet ~ Zones,
        data = subset_meta_Tourmalet %>% arrange(match(EHI_number,labels(neutral_Tourmalet))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_pop_Tourmalet_neut_1, warning=FALSE, comments="", message=FALSE}
pairwise <- pairwise.adonis(neutral_Tourmalet, subset_meta_Tourmalet$Zones, perm=999)
pairwise%>%
  tt()
```

***Phylogenetic***
```{r beta_div_pop_phylo_Tourmalet, warning=FALSE, comments="", message=FALSE}
phylo_Tourmalet<-as.matrix(beta_q1p$S)
phylo_Tourmalet <- as.dist(phylo_Tourmalet[rownames(phylo_Tourmalet) %in% samples_to_keep_Tourmalet,
               colnames(phylo_Tourmalet) %in% samples_to_keep_Tourmalet])
betadisper(phylo_Tourmalet, subset_meta_Tourmalet$Zones) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_phylo_Tourmalet, warning=FALSE, comments="", message=FALSE}
adonis2(phylo_Tourmalet ~ Zones,
        data = subset_meta_Tourmalet %>% arrange(match(EHI_number,labels(phylo_Tourmalet))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_pop_Tourmalet_phylo_1, warning=FALSE, comments="", message=FALSE}
pairwise <- pairwise.adonis(phylo_Tourmalet, subset_meta_Tourmalet$Zones, perm=999)
pairwise%>%
  tt()
```

***Functional***
```{r beta_div_pop_func_Tourmalet, warning=FALSE, comments="", message=FALSE}
func_Tourmalet<-as.matrix(beta_q0n$S)
func_Tourmalet <- as.dist(func_Tourmalet[rownames(func_Tourmalet) %in% samples_to_keep_Tourmalet,
               colnames(func_Tourmalet) %in% samples_to_keep_Tourmalet])
betadisper(func_Tourmalet, subset_meta_Tourmalet$Zones) %>% permutest(., pairwise = TRUE)
```

```{r beta_func_permanova_Tourmalet, warning=FALSE, comments="", message=FALSE}
adonis2(func_Tourmalet ~ Zones,
        data = subset_meta_Tourmalet %>% arrange(match(EHI_number,labels(func_Tourmalet))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_pop_Tourmalet_1_func, warning=FALSE, comments="", message=FALSE}
pairwise <- pairwise.adonis(func_Tourmalet, subset_meta_Tourmalet$Zones, perm=999)
pairwise%>%
  tt()
```

